---
title: "multiomics-cardiovascular-disease"
author: "Yan Luo"
date: '2024-08-16'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
# Make sure the working directory is "your path/multiomics-cardiovascular-disease/"
knitr::opts_knit$set(root.dir = "your path/multiomics-cardiovascular-disease/")

# data processing
library(skimr)
library(scales)
library(tidyverse)
library(tictoc)
library(data.table)
library(gtsummary)
library(jsonlite)

# visualization
library(ggplot2)
library(ggbreak)
library(ggpubr)
library(ggsci)
library(ggtext)
library(geomtextpath)
library(ggforce)
library(ggbeeswarm)
library(patchwork)
library(grid)
library(glue)
library(jjAnno)
library(MetBrewer)
library(paletteer)
library(survminer)
library(showtext)
library(scattermore)
library(TheBestColors)

# analysis
library(survival)
library(dcurves)
library(probably)
library(Hmisc)

# Self-defined functions
read_shap_data <- function(folder_path, outcome) {
  pattern <- paste0("^shap_", outcome, "_K[0-9]+\\.csv$")
  
  file_list <- list.files(path = folder_path, pattern = pattern, full.names = TRUE)
  
  data_list <- map(file_list, ~ {
    fold_value <- as.numeric(str_extract(.x, "(?<=K)\\d+"))
    df <- read_csv(.x, show_col_types = FALSE) %>%
      mutate(fold = fold_value, sample = row_number()) %>%
      select(fold, sample, everything())
    
    return(df)
  })
  
  return(data_list)
}

read_xtest_data <- function(folder_path) {
  pattern <- paste0("^xtest_K[0-9]+\\.csv$")
  
  file_list <- list.files(path = folder_path, pattern = pattern, full.names = TRUE)
  
  data_list <- map(file_list, ~ {
    fold_value <- as.numeric(str_extract(.x, "(?<=K)\\d+"))
    df <- read_csv(.x, show_col_types = FALSE) %>%
      mutate(fold = fold_value, sample = row_number()) %>%
      select(fold, sample, everything())
    
    return(df)
  })
  
  return(data_list)
}

add_extension_lines <- function(plot, n_rows, n_cols) {
  for (i in seq_len(n_rows + 1)) {
    plot <- plot + annotation_custom(
      grob = linesGrob(gp = gpar(lwd=1)), 
      xmin =-8.5, xmax = n_cols+0.5, ymin = i - 0.5, ymax = i - 0.5
    )
  }
  
  return(plot)
}

# Ref: https://dewey.dunnington.ca/post/2018/modifying-facet-scales-in-ggplot2/
FacetEqualWrap <- ggproto(
  "FacetEqualWrap", FacetWrap,
  
  train_scales = function(self, x_scales, y_scales, layout, data, params) {
    
    # doesn't make sense if there is not an x *and* y scale
    if (is.null(x_scales) || is.null(x_scales)) {
        stop("X and Y scales required for facet_equal_wrap")
    }
    
    # regular training of scales
    ggproto_parent(FacetWrap, self)$train_scales(x_scales, y_scales, layout, data, params)
    
    # switched training of scales (x and y and y on x)
    for (layer_data in data) {
      match_id <- match(layer_data$PANEL, layout$PANEL)
      
      x_vars <- intersect(x_scales[[1]]$aesthetics, names(layer_data))
      y_vars <- intersect(y_scales[[1]]$aesthetics, names(layer_data))
      
      SCALE_X <- layout$SCALE_X[match_id]
      ggplot2:::scale_apply(layer_data, y_vars, "train", SCALE_X, x_scales)
      
      SCALE_Y <- layout$SCALE_Y[match_id]
      ggplot2:::scale_apply(layer_data, x_vars, "train", SCALE_Y, y_scales)
    }
    
  }
)

facet_wrap_equal <- function(...) {
  # take advantage of the sanitizing that happens in facet_wrap
  facet_super <- facet_wrap(...)
  
  ggproto(NULL, FacetEqualWrap,
    shrink = facet_super$shrink,
    params = facet_super$params
  )
}

```

------------------------------------------------------------------------------------

# Read merged UK Biobank data

```{r}

print(getwd())

ukb_merged <- read_csv(str_c(getwd(), "/data/ukb_merged.csv"), show_col_types = F)

ukb_dict <- fromJSON(str_c(getwd(), "/data/ukb_dict.json"))

load(str_c(getwd(), "/data/nmr_info.rda"))

olink_assay <- read_delim(str_c(getwd(), "/data/olink_assay.dat.txt"), show_col_types = F, quote = "\t")

```


------------------------------------------------------------------------------------

# Descriptive statistics

## Supplementary Table 1: Clinical predictors

```{r}

dat <- ukb_merged %>%
  select(
    # Demographics
    age, male, ethnicity, townsend,
    # Lifestyles
    current_smoking, daily_drinking, healthy_sleep, physical_act, healthy_diet, social_active, 
    # Family history
    family_heart_hist, family_stroke_hist, family_hypt_hist, family_diab_hist,
    # Disease history and medication
    lipidlower, antihypt, hypt_hist, diab_hist,
    # Physical measurements
    sbp, dbp, height, weight, waist_cir, waist_hip_ratio, bmi,
    # Blood count
    wbc, lc, mc, nc, eos, baso, plt, hb, hct, 
    # Blood biochemistry
    apoA, apoB, total_cl, ldl_cl, hdl_cl, lpa, tg, glucose, hba1c, crt, cysc, urate, 
    urea, alb, tp, alt, ast, ggt, alp, dbil, tbil, crp, pho, ca, vd, igf1, shbg, trt
  ) %>%
  mutate(
    ethnicity = case_when(
      ethnicity == 0 ~ "White",
      ethnicity == 2 ~ "Asian",
      ethnicity == 3 ~ "Black",
      ethnicity %in% c(1,4) ~ "Others",
      TRUE ~ NA_character_
    ),
    ethnicity = factor(ethnicity, levels = c("White", "Black", "Asian", "Others")),
    current_smoking = ifelse(current_smoking == 1, "yes", "no"), current_smoking = relevel(factor(current_smoking), ref = "no"),
    daily_drinking = ifelse(daily_drinking == 1, "yes", "no"), daily_drinking = relevel(factor(daily_drinking), ref = "no"),
    healthy_sleep = ifelse(healthy_sleep == 1, "yes", "no"), healthy_sleep = relevel(factor(healthy_sleep), ref = "no"),
    physical_act = ifelse(physical_act == 1, "yes", "no"), physical_act = relevel(factor(physical_act), ref = "no"),
    healthy_diet = ifelse(healthy_diet == 1, "yes", "no"), healthy_diet = relevel(factor(healthy_diet), ref = "no"),
    social_active = ifelse(social_active == 1, "yes", "no"), social_active = relevel(factor(social_active), ref = "no"),
    family_heart_hist = ifelse(family_heart_hist == 1, "yes", "no"), family_heart_hist = relevel(factor(family_heart_hist), ref = "no"),
    family_stroke_hist = ifelse(family_stroke_hist == 1, "yes", "no"), family_stroke_hist = relevel(factor(family_stroke_hist), ref = "no"),
    family_hypt_hist = ifelse(family_hypt_hist == 1, "yes", "no"), family_hypt_hist = relevel(factor(family_hypt_hist), ref = "no"),
    family_diab_hist = ifelse(family_diab_hist == 1, "yes", "no"), family_diab_hist = relevel(factor(family_diab_hist), ref = "no"),
    lipidlower = ifelse(lipidlower == 1, "yes", "no"), lipidlower = relevel(factor(lipidlower), ref = "no"),
    antihypt = ifelse(antihypt == 1, "yes", "no"), antihypt = relevel(factor(antihypt), ref = "no"),
    hypt_hist = ifelse(hypt_hist == 1, "yes", "no"), hypt_hist = relevel(factor(hypt_hist), ref = "no"),
    diab_hist = ifelse(diab_hist == 1, "yes", "no"), diab_hist = relevel(factor(diab_hist), ref = "no")
  ) 


missing_perc <- tibble(
  variable = skim(dat)$skim_variable,
  missing_rate = (1 - skim(dat)$complete_rate)*100
)

var_need <- c("age", "male", "ethnicity", "townsend", "current_smoking", "daily_drinking", "healthy_sleep", "physical_act", "healthy_diet", "social_active", "family_heart_hist", "family_stroke_hist", "family_hypt_hist", "family_diab_hist", "lipidlower", "antihypt", "hypt_hist", "diab_hist", "sbp", "dbp", "height", "weight", "waist_cir", "waist_hip_ratio", "bmi", "wbc", "lc", "mc", "nc", "eos", "baso", "plt", "hb", "hct", "apoA", "apoB", "total_cl", "ldl_cl", "hdl_cl", "lpa", "tg", "glucose", "hba1c", "crt", "cysc", "urate", "urea", "alb", "tp", "alt", "ast", "ggt", "alp", "dbil", "tbil", "crp", "pho", "ca", "vd", "igf1", "shbg", "trt")

var_name <- c(
  "Age (years)",
  "Male",
  "Ethnicity",  
  "Townsend deprivation index",
  
  "Current smoking",
  "Daily alcohol intake",
  "Healthy sleep",
  "Physical activity",
  "Healthy diet",
  "Social connection",
  
  "Family history of heart disease",
  "Family history of stroke",
  "Family history of hypertension",
  "Family history of diabetes",
  
  "Lipid-lowering medication",
  "Antihypertensive medication",
  "Hypertension",
  "Diabetes",
  
  "Systolic blood pressure",
  "Diastolic blood pressure",
  "Height",
  "Weight",
  "Waist circumference",
  "Waist-hip ratio",
  "Body mass index",
  
  "Leukocyte count",
  "Lymphocyte count",
  "Monocyte count",
  "Neutrophill count",
  "Eosinophill count",
  "Basophill count",
  "Platelet count",
  "Haemoglobin concentration",
  "Haematocrit percentage",
  
  "Apolipoprotein A",
  "Apolipoprotein B",
  "Total cholesterol",
  "LDL cholesterol",
  "HDL cholesterol",
  "Lipoprotein(a)",
  "Triglycerides",
  "Glucose",
  "HbA1c",
  "Creatinine",
  "Cystatin C",
  "Urate",
  "Urea",
  "Albumin",
  "Total protein",
  "Alanine aminotransferase",
  "Aspartate aminotransferase",
  "Gamma glutamyl transferase",
  "Alkaline phosphatase",
  "Direct bilirubin",
  "Total bilirubin",
  "C-reactive protein",
  "Phosphate",
  "Calcium",
  "Vitamin D",
  "Insulin-like growth factor 1",
  "Sex hormone-binding globulin",
  "Testosterone"
)

label_list <- list()
for (i in 1:length(var_name)) {
  formula_str <- paste0(var_need[i], " ~ ", "\"", gsub("'", "'", var_name[i]), "\"")
  label_list[[i]] <- as.formula(formula_str)
}

tbl <- dat %>%
  tbl_summary(
    statistic = list(
      all_continuous() ~ "{median} ({p25}, {p75})",
      all_categorical() ~ "{n} ({p}%)"
    ),
    label = label_list,
    digits = list(all_continuous() ~ 1, all_categorical() ~ c(0, 1)),
    missing = "no"
    ) %>%
  modify_table_body(
    fun = ~.x %>%
      left_join(missing_perc, by = "variable") %>%
      mutate(
        missing_rate = case_when(
          row_type == "label" ~ missing_rate,
          TRUE ~ NA_real_
        )
      )
     ) %>%
  modify_header(
    label = "**Variable**",
    missing_rate = "**Missing rate (%)**"
    ) %>%
  modify_fmt_fun(missing_rate ~ function(x) style_number(x, digits = 1))

tbl %>%
  as_flex_table() %>%
  flextable::save_as_docx(path = str_c(getwd(), "/results/Table/Supplementary Table 1_Baseline characteristics.docx"))

rm(dat, var_need, var_name, label_list, tbl)

```

## Supplementary Table 11: Metabolomics

```{r}

met_list <- nmr_info %>% 
  filter(Nightingale == TRUE & Type %in% c("Non-derived", "Composite") & !Description %in% c("Spectrometer-corrected alanine", "Glucose-lactate")) %>%
  pull(Biomarker)

dat <- ukb_merged %>% 
  select(all_of(met_list))

summary_stats <- dat %>%
  summarise(across(everything(), 
                   list(
                     missing = ~mean(is.na(.)) * 100,
                     median = ~median(., na.rm = TRUE),
                     Q25 = ~quantile(., 0.25, na.rm = TRUE),
                     Q75 = ~quantile(., 0.75, na.rm = TRUE)
                   ), .names = "{col}_{fn}")) %>%
  pivot_longer(cols = everything(),
               names_to = c("variable", "statistic"),
               names_pattern = "(.*)_(.*)",
               values_to = "value") %>%
  pivot_wider(names_from = "statistic", values_from = "value") %>%
  mutate(
    median_iqr = sprintf("%0.3f (%0.3f, %0.3f)", median, Q25, Q75),
    missing_rate = missing
  ) %>% 
  select(variable, median_iqr, missing_rate)

subgroup_order <- c(
  "Amino acids",
  "Branched-chain amino acids",
  "Aromatic amino acids",
  "Fluid balance",
  "Inflammation",
  "Fatty acids",
  "Glycolysis related metabolites",
  "Ketone bodies",
  
  "Total lipids",
  "Cholesterol",
  "Free cholesterol",
  "Cholesteryl esters",
  "Phospholipids",
  "Triglycerides",
  "Other lipids",
 
  "Lipoprotein particle sizes",
  "Lipoprotein particle concentrations",
  "Chylomicrons and extremely large VLDL",
  "Very large VLDL",
  "Large VLDL",
  "Medium VLDL",
  "Small VLDL",
  "Very small VLDL",

  "Large LDL",
  "Medium LDL",
  "Small LDL",
  "IDL",
  "Very large HDL",
  "Large HDL",
  "Medium HDL",
  "Small HDL",
  "Apolipoproteins"
  )

metabolites <- nmr_info %>% 
  filter(Nightingale == TRUE & Type %in% c("Non-derived", "Composite") & !Description %in% c("Spectrometer-corrected alanine", "Glucose-lactate")) %>%
  mutate(
    Sub.Group = ifelse(Sub.Group == "", Group, Sub.Group),
    Sub.Group = sub("\\s*\\(.*\\)", "", Sub.Group),
    Sub.Group = factor(Sub.Group, levels = subgroup_order)
  ) %>%
  select(Biomarker, Description, Units, Sub.Group, UKB.Field.ID) %>%
  arrange(Sub.Group, Biomarker) %>%
  left_join(summary_stats, by = c("Biomarker" = "variable"))

write.csv(metabolites, "your path/results/Table/Supplemetanry Table 11_metabolomics.csv", row.names = F)

rm(dat, met_list, summary_stats, subgroup_order, metabolites)

```

## Supplementary Table 12: Proteomics

```{r}

pro_list <- olink_assay %>% 
  filter(!Assay %in% c("GLIPR1", "NPM1", "PCOLCE")) %>%
  pull(Assay)

dat <- ukb_merged %>% 
  select(all_of(pro_list))

summary_stats <- dat %>%
  summarise(across(everything(), 
                   list(
                     missing = ~mean(is.na(.)) * 100,
                     median = ~median(., na.rm = TRUE),
                     Q25 = ~quantile(., 0.25, na.rm = TRUE),
                     Q75 = ~quantile(., 0.75, na.rm = TRUE)
                   ), .names = "{col}_{fn}")) %>%
  pivot_longer(cols = everything(),
               names_to = c("variable", "statistic"),
               names_pattern = "(.*)_(.*)",
               values_to = "value") %>%
  pivot_wider(names_from = "statistic", values_from = "value") %>%
  mutate(
    median_iqr = sprintf("%0.3f (%0.3f, %0.3f)", median, Q25, Q75),
    missing_rate = missing
  ) %>% 
  select(variable, median_iqr, missing_rate)

proteins <- olink_assay %>% 
  filter(!Assay %in% c("GLIPR1", "NPM1", "PCOLCE")) %>%
  select(Assay, Panel) %>%
  arrange(Panel, Assay) %>%
  left_join(summary_stats, by = c("Assay" = "variable"))

write.csv(proteins, "your path/results/Table/Supplemetanry Table 12_proteomics.csv", row.names = F)

rm(dat, pro_list, summary_stats, proteins)

```


------------------------------------------------------------------------------------

# Supplementary Fig.1: CVD event distribution + Followup information + CVD-free survival

## CVD event distribution

```{r}

outcomes_list <- c("cved", "cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")

outcomes_name <- c("All cardiovascular outcomes", "Coronary artery disease", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery disease", "Abdominal aortic aneurysm", "Venous thromboembolism", "Cardiovascular death")

#### Data preparation ####
dat <- ukb_merged %>%
  # Step 1: Pivot to get 'yrs' and 'outcome_yrs' from bl2 columns
  select(eid, matches("^(eid|cved|mace|cad|stroke|hf|af|va|pad|aaa|vt|cvd_death)$|^bl2(cved|mace|cad|stroke|hf|af|va|pad|aaa|vt|cvd_death)_yrs$")) %>%
  pivot_longer(cols = starts_with("bl2"), names_to = "outcome_yrs", values_to = "yrs") %>%
  mutate(outcome_yrs = sub("bl2", "", outcome_yrs)) %>%
  
  # Step 2: Pivot again to get 'value' and 'outcome_value' for each outcome
  pivot_longer(
    cols = matches("^(cved|mace|cad|stroke|hf|af|va|pad|aaa|vt|cvd_death)$"), 
    names_to = "outcome_value", values_to = "value") %>%
  
  # Step 3: Ensure matching outcomes for both pivots
  filter(str_detect(outcome_yrs, outcome_value)) %>%
  
  # Step 4: Select and rename columns
  select(eid, outcome_value, value, yrs) %>%
  rename(outcome = outcome_value) %>%
  
  # Step 5: Filter for outcomes of interest
  filter(outcome %in% outcomes_list & value == 1) %>%
  mutate(
    outcome = factor(outcome, levels = rev(outcomes_list), labels = rev(outcomes_name))
  )

case_counts <- dat %>%
  group_by(outcome) %>%
  summarise(cases = sum(value)) %>%
  ungroup() %>%
  mutate(
    rates = cases / nrow(dat) * 100,
    cases_rates = sprintf("%0.0f (%0.1f%%)", cases, rates)
    )

# color
scales::show_col(pal_npg("nrc", alpha = 0.8)(10)) # with "#F2C76C"

#### Plotting ####
# Scatter plot with jitter
p1 <- ggplot(dat, aes(x = yrs, y = outcome, color = outcome)) +
  geom_point(
    alpha = 0.6, size = 2, 
    position = position_jitter(width = 0, height = 0.1, seed = 97)
    ) +
  scale_colour_manual(
    values = rev(c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#F2C76C", "#91D1C2CC", "#DC0000CC", "#7E6148CC"))
  ) +
  theme_bw() +
  theme(
    panel.grid = element_blank(),
    # panel.border = element_rect(color = "black", linewidth = 0.75, fill = NA),
    axis.text.x = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5)
  ) +
  labs(
    x = "Observation time (years)",
    y = NULL,
    title = NULL
  )

# Lollipop chart
p2 <- ggplot(case_counts, aes(x = cases, y = outcome)) +
  geom_segment(aes(x = 0, xend = cases, y = outcome, yend = outcome, color = outcome)) +
  geom_point(aes(color = outcome), size = 3) +
  geom_text(aes(label = cases_rates, hjust = -0.15), size = 3, color = "black") +
  scale_colour_manual(
    values = rev(c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#F2C76C", "#91D1C2CC", "#DC0000CC", "#7E6148CC"))
  ) +
  scale_x_continuous(
    expand = c(0, 0), limits = c(0, 7250), breaks = c(0, 2000, 4000, 6000)
    ) +
  theme_bw() +
  theme(
    panel.grid = element_blank(),
    axis.text.x = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5)
  ) +
  labs(
    x = "Number of cases",
    y = NULL,
    title = NULL
  )

# Pool 
scatter_lollipop <- wrap_elements(
  p1 + p2 + plot_layout(ncol = 2, widths = c(1, 0.3))
)

```

## Followup information

```{r}

#### Data preparation ####
# Median follow-up time
median_follow_up_time <- median(ukb_merged$bl2cved_yrs, na.rm = TRUE)
round(quantile(ukb_merged$bl2cved_yrs, seq(0, 1, 0.25)), 1)
sum(ukb_merged$bl2cved_yrs) # person-years


#### Plotting ####
density <- ggplot(ukb_merged, aes(x = bl2cved_yrs)) +
  geom_density(fill = "#8491B4", alpha = 0.6) + 
  annotate("segment", x = median_follow_up_time, y = 0, yend = 0.28, color = "#696969", linetype = "dashed", linewidth = 1) +
  annotate("text", x = median_follow_up_time - 6, y = 0.04, label = paste0("Median: ", round(median_follow_up_time, 1), " years"), color = "black", size = 5, hjust = 0) +
  labs(
    x = "Observation time (years)",
    y = "Density"
  ) +
  theme_classic() +
  theme(
    panel.grid = element_blank(),
    axis.text.x = element_text(size = 12, colour = "black"),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank()
  ) + 
  scale_y_continuous(
    expand = c(0, 0), limits = c(0, 0.35)
    )

```

## CVD-free survival

```{r}

#### Data preparation ####
cvd_free_survival <- read_csv(str_c(getwd(), "/data/cvd_free_survival.csv"), show_col_types = F)

# color
scales::show_col(pal_npg("nrc", alpha = 0.8)(2))

#### Plotting ####
surv_plot <- ggplot(cvd_free_survival, aes(x = timeline, y = survival_prob, color = sex)) +
  geom_line(linewidth = 0.5) + 
  geom_ribbon(
    aes(ymin = lower_0.95, ymax = upper_0.95, fill = sex), 
    alpha = 0.2, colour = "white") + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black"),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    legend.position = "inside",
    legend.position.inside = c(0.9, 0.85)
  ) +
  scale_y_continuous(limits = c(0.65, 1.05), breaks = c(0.7, 0.8, 0.9, 1)) +
  scale_colour_manual(values = c("#E64B35CC", "#4DBBD5CC"), name = NULL) +
  scale_fill_manual(values = c("#E64B35CC", "#4DBBD5CC"), name = NULL) +
  labs(
    x = "Observation time (years)",
    y = "Disease-free survival probability"
  )

```

## Pool

```{r}

design <- "
AAABBB
CCCCCC
CCCCCC
"

cvd_followup_survival <- wrap_plots(
  density, surv_plot, scatter_lollipop, 
  design = design
) +
  plot_annotation(tag_levels = "a") & 
  theme(plot.tag = element_text(face = "bold"))

ggsave(str_c(getwd(), "/results/Figure/CVD_followup_survival.tiff"), plot = cvd_followup_survival, width = 12, height = 9, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/CVD_followup_survival.png"), plot = cvd_followup_survival, width = 12, height = 9, unit = "in", dpi = 300, device = "png")

ggsave(str_c(getwd(), "/results/Figure/CVD_followup_survival.pdf"), plot = cvd_followup_survival, width = 12, height = 9, unit = "in", dpi = 300, device = "pdf")

```

------------------------------------------------------------------------------------

# Supplementary Fig.4: Observed event rates 

```{r}

#### Data preparation ####
observed_event_rate <- read_csv(str_c(getwd(), "/data/observed_event_rate.csv"), show_col_types = F)


dat_prs <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for(i in 1:9){
  prs <- observed_event_rate %>%
    filter(outcome == outcomes[i]) %>%
    pull(prs)
  
  for(j in 0:99){
    lb_prs <- quantile(prs, j/100)
    up_prs <- quantile(prs, (j+1)/100)
    
    dat <- observed_event_rate %>%
      filter(outcome == outcomes[i] & prs >= lb_prs & prs < up_prs) %>%
      mutate(
        event_rate = sum(event) / n(),
        quantile = j+1
      ) %>%
      slice(1) %>%
      select(outcome, event_rate, quantile)
    
    dat_prs <- bind_rows(dat_prs, dat)
  }
  rm(lb_prs, up_prs, dat, j, i, prs)
}

dat_met <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for(i in 1:9){
  metscore <- observed_event_rate %>%
    filter(outcome == outcomes[i]) %>%
    pull(metscore)
  
  for(j in 0:99){
    lb_met <- quantile(metscore, j/100)
    up_met <- quantile(metscore, (j+1)/100)
    
    dat <- observed_event_rate %>%
      filter(outcome == outcomes[i] & metscore >= lb_met & metscore < up_met) %>%
      mutate(
        event_rate = sum(event) / n(),
        quantile = j+1
      ) %>%
      slice(1) %>%
      select(outcome, event_rate, quantile)
    
    dat_met <- bind_rows(dat_met, dat)
  }
  rm(lb_met, up_met, dat, j, i, metscore)
}


dat_pro <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for(i in 1:9){
  proscore <- observed_event_rate %>%
    filter(outcome == outcomes[i]) %>%
    pull(proscore)
  
  for(j in 0:99){
    lb_pro <- quantile(proscore, j/100)
    up_pro <- quantile(proscore, (j+1)/100)
    
    dat <- observed_event_rate %>%
      filter(outcome == outcomes[i] & proscore >= lb_pro & proscore < up_pro) %>%
      mutate(
        event_rate = sum(event) / n(),
        quantile = j+1
      ) %>%
      slice(1) %>%
      select(outcome, event_rate, quantile)
    
    dat_pro <- bind_rows(dat_pro, dat)
  }
  rm(lb_pro, up_pro, dat, j, i, proscore)
}

dat <- bind_rows(
  dat_prs %>% mutate(score = "PRS"), 
  dat_met %>% mutate(score = "MetScore"), 
  dat_pro %>% mutate(score = "ProScore")) %>%
  mutate(
    score = factor(score, levels = c("PRS", "MetScore", "ProScore")),
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
# Best100(50)[5]
# Best100(57)[5]
# Best100(52)[5]
# scales::show_col(c("#74C476", "#9E9AC8", "#FD8D3C"))

#### Plotting ##### 初始化一个空的数据框
cor <- data.frame(score = character(), outcome = character(), rho = numeric(), pvalue = numeric(), stringsAsFactors = FALSE)

scores <- c("PRS", "Met", "Pro")
data_list <- list(dat_prs, dat_met, dat_pro)

for(j in 1:length(scores)) {
  for(i in outcomes) {
    print(i)
    cor_test_result <- cor.test(
      data_list[[j]] %>% filter(outcome == i) %>% pull(quantile),
      data_list[[j]] %>% filter(outcome == i) %>% pull(event_rate),
      method = "spearman"
    )
    
    cor <- rbind(cor, data.frame(
      score = scores[j],
      outcome = i,
      rho = cor_test_result$estimate,
      pvalue = cor_test_result$p.value
    ))
  }
}

cor <- cor %>% 
  mutate(
    rho_text = sprintf("%0.2f", rho),
  )

for(i in outcomes){
  print(i)
  print(cor.test(dat_prs %>% filter(outcome == i) %>% pull(quantile), dat_prs %>% filter(outcome == i) %>% pull(event_rate), method = "spearman"))
}

scatter_plot <- ggplot(dat, aes(x = quantile, y = event_rate*100, color = score)) +
  geom_point(size = 2.5, alpha = 0.7) +
  theme_classic() + 
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0),
    axis.title.y = element_text(size = 12, colour = "black"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0.1, b = 0, l = 0.1), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_color_manual(values = c("PRS" = "#66C2A4", "MetScore" = "#9E9AC8", "ProScore" = "#FD8D3C")) +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Predicted score percentile (%)", 
    y = "Observed event rate (%)", 
    color = "Score")


scatter_plot_log <- ggplot(dat, aes(x = quantile, y = log(event_rate*100), color = score)) +
  geom_point(size = 2.5) +
  theme_classic() + 
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0),
    axis.title.y = element_text(size = 12, colour = "black"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0.1, b = 0, l = 0.1), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_color_manual(values = c("PRS" = "#66C2A4", "MetScore" = "#9E9AC8", "ProScore" = "#FD8D3C")) +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Predicted score percentile (%)", 
    y = "Observed event rate (%)", 
    color = "Score")

ggsave(str_c(getwd(), "/results/Figure/Observed_events.pdf"), plot = scatter_plot, width = 12, height = 7, unit = "in", dpi = 500, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Observed_events.tiff"), plot = scatter_plot, width = 12, height = 7, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Observed_events.png"), plot = scatter_plot, width = 12, height = 7, unit = "in", dpi = 300, device = "png")

```

# Survival probability by PRS, MetScore and ProScore

## Kaplan-Meier curve

### Supplementary Fig.3: PRS

```{r}

#### Data preparation ####
survival_prs <- read_csv(str_c(getwd(), "/data/survival_prs.csv"), show_col_types = F) %>%
  mutate(
    group = factor(group, levels = c("low", "medium", "high"), labels = c("Low", "Medium", "High")),
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
# Best100(50)[c(4, 6, 9)]
# "#A1D99B", "#41AB5D", "#00441B"

#### Plotting ####
surv_prs <- ggplot(
  survival_prs, 
  aes(x = timeline, y = survival_prob*100, color = group)) +
  geom_line(linewidth = 0.5) + 
  geom_ribbon(
    aes(ymin = lower_0.95*100, ymax = upper_0.95*100, fill = group), 
    alpha = 0.2, colour = "white") + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black"),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.grid = element_blank(),
    panel.spacing = unit(1, "lines")
  ) +
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  scale_colour_manual(
    values = c("#A1D99B", "#41AB5D", "#00441B"), 
    name = "PRS") +
  scale_fill_manual(
    values = c("#A1D99B", "#41AB5D", "#00441B"), 
    name = "PRS") +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Observation time (years)",
    y = "Survival probability (%)"
  )

ggsave(str_c(getwd(), "/results/Figure/Survival_prob_PRS.pdf"), plot = surv_prs, width = 12, height = 7, unit = "in", dpi = 500, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Survival_prob_PRS.tiff"), plot = surv_prs, width = 12, height = 7, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Survival_prob_PRS.png"), plot = surv_prs, width = 12, height = 7, unit = "in", dpi = 300, device = "png")

```

### MetScore

```{r}

#### Data preparation ####
survival_met <- read_csv(str_c(getwd(), "/data/survival_met.csv"), show_col_types = F) %>%
  mutate(
    group = factor(group, levels = c("low", "medium", "high"), labels = c("Low", "Medium", "High")),
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
# Best100(57)[c(4, 6, 9)]
# "#BCBDDC", "#807DBA", "#3F007D"

#### Plotting ####
surv_met <- ggplot(
  survival_met, 
  aes(x = timeline, y = survival_prob*100, color = group)) +
  geom_line(linewidth = 0.5) + 
  geom_ribbon(
    aes(ymin = lower_0.95*100, ymax = upper_0.95*100, fill = group), 
    alpha = 0.2, colour = "white") + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black"),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.grid = element_blank(),
    panel.spacing = unit(1, "lines")
  ) +
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  scale_colour_manual(
    values = c("#BCBDDC", "#807DBA", "#3F007D"), 
    name = "MetScore") +
  scale_fill_manual(
    values = c("#BCBDDC", "#807DBA", "#3F007D"), 
    name = "MetScore") +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Observation time (years)",
    y = "Survival probability (%)"
  )

```

### ProScore

```{r}

#### Data preparation ####
survival_pro <- read_csv(str_c(getwd(), "/data/survival_pro.csv"), show_col_types = F) %>%
  mutate(
    group = factor(group, levels = c("low", "medium", "high"), labels = c("Low", "Medium", "High")),
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
# Best100(52)[c(4, 6, 9)]
# "#FDAE6B", "#F16913", "#7F2704"


#### Plotting ####
surv_pro <- ggplot(
  survival_pro, 
  aes(x = timeline, y = survival_prob*100, color = group)) +
  geom_line(linewidth = 0.5) + 
  geom_ribbon(
    aes(ymin = lower_0.95*100, ymax = upper_0.95*100, fill = group), 
    alpha = 0.2, colour = "white") + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black"),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12),
    axis.title.y = element_text(size = 12),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.grid = element_blank(),
    panel.spacing = unit(1, "lines")
  ) +
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  scale_colour_manual(
    values = c("#FDAE6B", "#F16913", "#7F2704"), 
    name = "ProScore") +
  scale_fill_manual(
    values = c("#FDAE6B", "#F16913", "#7F2704"), 
    name = "ProScore") +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Observation time (years)",
    y = "Survival probability (%)"
  )

```

### Fig.2: Pool

```{r}

surv_pool <- wrap_plots(
   surv_met + surv_pro +
    plot_layout(nrow = 2, heights = c(1, 1))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))


ggsave(str_c(getwd(), "/results/Figure/Survival_prob.pdf"), plot = surv_pool, width = 12, height = 14, unit = "in", dpi = 500, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Survival_prob.tiff"), plot = surv_pool, width = 12, height = 14, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Survival_prob.png"), plot = surv_pool, width = 12, height = 14, unit = "in", dpi = 300, device = "png")


```


## Supplementary Fig.5: Hazard ratio

```{r}

#### Data preparation ####
hrs_prs <- read_csv(str_c(getwd(), "/data/survival_hazard_ratio_prs.csv"), show_col_types = F)

hrs_met <- read_csv(str_c(getwd(), "/data/survival_hazard_ratio_met.csv"), show_col_types = F)

hrs_pro <- read_csv(str_c(getwd(), "/data/survival_hazard_ratio_pro.csv"), show_col_types = F)

dat <- bind_rows(
  hrs_prs %>% mutate(score = "PRS"), 
  hrs_met %>% mutate(score = "MetScore"), 
  hrs_pro %>% mutate(score = "ProScore")) %>%
  mutate(
    score = factor(score, levels = c("PRS", "MetScore", "ProScore")),
    covariate = case_when(
      str_detect(covariate, "_high") ~ "High",
      str_detect(covariate, "_medium") ~ "Medium"
    ),
    covariate = factor(covariate, levels = c("Medium", "High")),
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = case_when(
      score == "PRS" & covariate == "Medium" ~ "#41AB5D",
      score == "PRS" & covariate == "High" ~ "#00441B",
      score == "MetScore" & covariate == "Medium" ~ "#807DBA",
      score == "MetScore" & covariate == "High" ~ "#3F007D",
      score == "ProScore" & covariate == "Medium" ~ "#F16913",
      score == "ProScore" & covariate == "High" ~ "#7F2704"
    ),
    hr = exp(coef),
    lower_ci = `exp(coef) lower 95%`,
    upper_ci = `exp(coef) upper 95%`,
    hr_ci = sprintf("%0.2f [%0.2f–%0.2f]", hr, lower_ci, upper_ci)
  ) %>%
  select(hr_ci, score, outcome, everything())

#### Plotting ####
hrs_forest_plot <- ggplot(
  dat, aes(x = covariate, y = hr, group = score, color = score)
  ) +
  geom_point(size = 2, alpha = 1, position = position_dodge(width = 0.5)) +
  geom_linerange(aes(ymin = lower_ci, ymax = upper_ci), linewidth = 1, position = position_dodge(width = 0.5)) + 
  geom_hline(yintercept = 1, linetype = "dashed") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0),
    axis.title.y = element_text(size = 12, colour = "black"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0.5, b = 0, l = 0.1), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_color_manual(values = c("PRS" = "#66C2A4", "MetScore" = "#9E9AC8", "ProScore" = "#FD8D3C")) +
  coord_flip() +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = NULL, 
    y = "Hazard ratio", 
    color = "Score")

ggsave(str_c(getwd(), "/results/Figure/Hazard_ratio_forest_plot.pdf"), plot = hrs_forest_plot, width = 10, height = 7, unit = "in", dpi = 500, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Hazard_ratio_forest_plot.tiff"), plot = hrs_forest_plot, width = 10, height = 7, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Hazard_ratio_forest_plot.png"), plot = hrs_forest_plot, width = 10, height = 7, unit = "in", dpi = 300, device = "png")

```


------------------------------------------------------------------------------------

# Descriptive statistics of omics scores

## Distribution

```{r}

#### Data preparation ####
# MetScore
file_paths <- str_c(getwd(), "/data/MetScore/test_scores_K", 0:9, ".csv")
summary_metscore <- map2_dfr(
  file_paths,
  .y = 0:9,
  .f = function(file, idx) {
    df <- read_csv(file, show_col_types = FALSE)

    df %>%
      select(-eid) %>%
      summarise(across(everything(), list(mean = mean, sd = sd), na.rm = TRUE)) %>%
      pivot_longer(
        cols = everything(),
        names_to = c("variable", ".value"),
        names_pattern = "^(.*)_(mean|sd)$"
      ) %>%
      mutate(dataset = str_c("K", idx))
    }
  ) 
summary_metscore <- summary_metscore %>%
  mutate(
    variable = factor(
      variable, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
    )

# ProScore
file_paths <- str_c(getwd(), "/data/ProScore/test_scores_K", 0:9, ".csv")
summary_proscore <- map2_dfr(
  file_paths,
  .y = 0:9,
  .f = function(file, idx) {
    df <- read_csv(file, show_col_types = FALSE)

    df %>%
      select(-eid) %>%
      summarise(across(everything(), list(mean = mean, sd = sd), na.rm = TRUE)) %>%
      pivot_longer(
        cols = everything(),
        names_to = c("variable", ".value"),
        names_pattern = "^(.*)_(mean|sd)$"
      ) %>%
      mutate(dataset = str_c("K", idx))
    }
  )
summary_proscore <- summary_proscore %>%
  mutate(
    variable = factor(
      variable, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
    )


#### Plotting ####
bar_plot_metscore <- ggplot(summary_metscore, aes(x = dataset, y = mean)) +
  geom_bar(stat = "identity", fill = "#9E9AC8") +
  geom_errorbar(aes(ymin = mean - sd, ymax = mean + sd), width = 0.2) +
  facet_wrap(~ variable, scales = "free_y") +
  theme_bw() +
  labs(
    x = "Cross-validation dataset",
    y = "Mean (± SD)",
    title = "a MetScore"
  ) +
  theme(
    strip.text = element_text(size = 12),
    axis.text.x = element_text(hjust = 1),
    plot.title = element_text(size = 14, face = "bold")
  )

bar_plot_proscore <- ggplot(summary_proscore, aes(x = dataset, y = mean)) +
  geom_bar(stat = "identity", fill = "#FD8D3C") +
  geom_errorbar(aes(ymin = mean - sd, ymax = mean + sd), width = 0.2) +
  facet_wrap(~ variable, scales = "free_y") +
  theme_bw() +
  labs(
    x = "Cross-validation dataset",
    y = "Mean (± SD)",
    title = "b ProScore"
  ) +
  theme(
    strip.text = element_text(size = 12),
    axis.text.x = element_text(hjust = 1),
    plot.title = element_text(size = 14, face = "bold")
  )

bar_plot_pool <- bar_plot_metscore / bar_plot_proscore


ggsave(str_c(getwd(), "/results/Figure/Omics_score_bar_plot.pdf"), plot = bar_plot_pool, width = 10, height = 14, unit = "in", dpi = 500, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Omics_score_bar_plot.tiff"), plot = bar_plot_pool, width = 10, height = 14, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Omics_score_bar_plot.png"), plot = bar_plot_pool, width = 10, height = 14, unit = "in", dpi = 300, device = "png")

```

## Supplementary Fig.2: Correlation

```{r}

#### Data preparation ####
# MetScore
file_paths <- str_c(getwd(), "/data/MetScore/test_scores_K", 0:9, ".csv")
cor_metscore <- map_dfr(
  file_paths,
  .f = function(file) {
    df <- read_csv(file, show_col_types = FALSE)
    df %>% select(-eid)
    }
  ) %>%
  cor(method = "spearman", use = "pairwise.complete.obs") %>%
  as.data.frame() %>%
  rownames_to_column("var1") %>%
  pivot_longer(cols = -var1, names_to = "var2", values_to = "cor") %>%
  mutate(
    var1 = factor(
      var1, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    var2 = factor(
      var2, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
    )

# ProScore
file_paths <- str_c(getwd(), "/data/ProScore/test_scores_K", 0:9, ".csv")
cor_proscore <- map_dfr(
  file_paths,
  .f = function(file) {
    df <- read_csv(file, show_col_types = FALSE)
    df %>% select(-eid)
    }
  ) %>%
  cor(method = "spearman", use = "pairwise.complete.obs") %>%
  as.data.frame() %>%
  rownames_to_column("var1") %>%
  pivot_longer(cols = -var1, names_to = "var2", values_to = "cor") %>%
  mutate(
    var1 = factor(
      var1, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    var2 = factor(
      var2, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
    )


#### Plotting ####
cor_plot_metscore <- ggplot(cor_metscore, aes(var1, var2, fill = cor)) +
  geom_tile(color = "white", linewidth = 0.5) +
  scale_fill_gradientn(
    colours = c("#2E5A87", "#7BA8CCFF", "#FFFFFFFF", "#F78171FF", "#A90C38"),
    breaks = c(0.7, 0.8, 0.9, 1),
    limit = c(0.7, 1),
    space = "Lab",
    name = "Spearman correlation") +
  theme_void()+ 
  theme(
    axis.text.y = element_text(size = 12, hjust = 1, margin = margin(r = 5)),
    axis.text.x = element_text(size = 12, margin = margin(t = 6)),
    axis.title.y = element_blank(),
    axis.title.x = element_blank(),
    axis.ticks = element_line(color = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.15, "cm"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 12),
    legend.position = "right",
    plot.margin = unit(c(t = 0, r = .5, b = 0, l = .5),"cm"),
    plot.title = element_text(size = 14, face = "bold", hjust = 0, vjust = 1)
    ) +
  labs(
    x = "",
    y = "",
    title = "a MetScore"
  )

cor_plot_proscore <- ggplot(cor_proscore, aes(var1, var2, fill = cor)) +
  geom_tile(color = "white", linewidth = 0.5) +
  scale_fill_gradientn(
    colours = c("#2E5A87", "#7BA8CCFF", "#FFFFFFFF", "#F78171FF", "#A90C38"),
    breaks = c(0.7, 0.8, 0.9, 1),
    limit = c(0.7, 1),
    space = "Lab",
    name = "Spearman correlation") +
  theme_void()+ 
  theme(
    axis.text.y = element_text(size = 12, hjust = 1, margin = margin(r = 5)),
    axis.text.x = element_text(size = 12, margin = margin(t = 6)),
    axis.title.y = element_blank(),
    axis.title.x = element_blank(),
    axis.ticks = element_line(color = "black", linewidth = 0.5),
    axis.ticks.length = unit(0.15, "cm"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 12),
    legend.position = "right",
    plot.margin = unit(c(t = 0, r = .5, b = 0, l = .5),"cm"),
    plot.title = element_text(size = 14, face = "bold", hjust = 0, vjust = 1)
    ) +
  labs(
    x = "",
    y = "",
    title = "b ProScore"
  )

cor_plot_pool <- cor_plot_metscore + cor_plot_proscore +
  plot_layout(guides = "collect")

ggsave(str_c(getwd(), "/results/Figure/Omics_score_correlation_plot.pdf"), plot = cor_plot_pool, width = 14, height = 6, unit = "in", dpi = 500, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Omics_score_correlation_plot.tiff"), plot = cor_plot_pool, width = 14, height = 6, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Omics_score_correlation_plot.png"), plot = cor_plot_pool, width = 14, height = 6, unit = "in", dpi = 300, device = "png")

```


------------------------------------------------------------------------------------

# Fig.3: C-index comparison

## C-index forest plot

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro")), labels = c("ProScore", "MetScore", "PRS")),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

dat_text <- dat %>% 
  arrange(model, desc(mean_cindex)) %>% 
  mutate(
    cindex_text = sprintf("%0.2f (%0.2f–%0.2f)", mean_cindex, cindex_2_5, cindex_97_5)
  )


# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Clinical scores

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("ASCVD", "SCORE2")) %>%
  mutate(
    model = factor(model, levels = rev(c("ASCVD", "SCORE2"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

dat_text <- dat %>% 
  arrange(model, desc(mean_cindex)) %>% 
  mutate(
    cindex_text = sprintf("%0.2f (%0.2f–%0.2f)", mean_cindex, cindex_2_5, cindex_97_5)
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[2]


#### Plotting ####
f1_scores <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_scores <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_scores <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_scores <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_scores <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_scores <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_scores <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_scores <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_scores <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#4DBBD5CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#4DBBD5CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CAD" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "Stroke" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "HF" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "AF" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "VA" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "PAD" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "AAA" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "VTE" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CVDeath" & model == "AgeSex") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CAD" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "Stroke" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "HF" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "AF" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "VA" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "PAD" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "AAA" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "VTE" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CVDeath" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

#### Example for Figure 1

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )


#### Plotting ####
example <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CAD" & model == "Clin") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0.5, r = 0.5, b = 0.5, l = 0.5), "cm")
  ) +
  labs(
    x = NULL,
    y = "C-index",
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.81), breaks = c(0.5, 0.8), labels = c("0.5", "1.0")) +
  coord_flip()

ggsave(str_c(getwd(), "/results/Figure/Cindex_comparison_example.tiff"), plot = example, width = 4, height = 3, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Cindex_comparison_example.png"), plot = example, width = 4, height = 3, unit = "in", dpi = 300, device = "png")

```


### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CAD" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "Stroke" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "HF" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "AF" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "VA" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "PAD" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "AAA" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "VTE" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_hline(
    yintercept = dat %>% filter(outcome == "CVDeath" & model == "PANEL") %>% pull(mean_cindex),
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_scores + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_scores + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_scores + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_scores + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_scores + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_scores + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_scores + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_scores + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_scores + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 5, heights = c(3/8, 2/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_pool <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_CPH.png"), plot = cindex_forest_pool, width = 16, height = 10, unit = "in", dpi = 300, device = "png")


#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clinical scores
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("ASCVD", "SCORE2")) %>%
  mutate(
    model = factor(model, levels = rev(c("ASCVD", "SCORE2"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clinical score"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex.tsv"))

# Used for comparison with Lasso
cindex_by_model_cph <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex)) 

```

## Delta C-index

### AgeSex-based

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "AgeSex"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model == "AgeSex", "white", "#00A087CC")
  )


dat_agesex <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0) %>% 
  mutate(
    delta_cindex_text = sprintf("%0.3f (%0.3f–%0.3f)", mean_delta_cindex, delta_cindex_2_5, delta_cindex_97_5)
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

### Clinical-based

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "Clin"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model == "Clin", "white", "#3C5488CC")
  )

dat_sig <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0)

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

### PANEL-based

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANEL"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model == "PANEL", "white", "#F39B7FCC")
  )

dat_sig <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0)

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 8, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
delta_forest_pool <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    caption = "\u0394 C-index",
    theme = theme(
      plot.caption = element_markdown(size = 14, hjust = 0.5, vjust = 1, family = "Arial")
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


#### Source data ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

delta_cindex <- tibble()

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "AgeSex"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "Clin"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANEL"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

write_tsv(delta_cindex, file = str_c(getwd(), "/results/Table/Source data/delta_cindex.tsv"))

```

## Pool

```{r}

# Legend
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c("PRS", "Met", "Pro", "ASCVD", "SCORE2", "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro", "Clin", "ClinPRS", "ClinMet", "ClinPro", "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro", "PANEL", "PANELPRS", "PANELMet", "PANELPro", "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    group = case_when(
      model %in% c("PRS", "Met", "Pro") ~ "Omics-based",
      model %in% c("ASCVD", "SCORE2") ~ "Clinical scores",
      str_detect(model, "AgeSex") ~ "AgeSex-based",
      str_detect(model, "Clin") ~ "Clinical-based",
      str_detect(model, "PANEL") ~ "PANEL-based"
    ),
    group = factor(group, levels = c("Omics-based", "Clinical scores", "AgeSex-based", "Clinical-based", "PANEL-based")),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

legend <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex, colour = group)
  ) +
  geom_point(size = 2, alpha = 1) +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1) + 
  theme_light() +
  theme(
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.title = element_text(size = 12, hjust = .5)
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.4, 1)) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC")
  )

legend <- wrap_elements(ggpubr::get_legend(legend)) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


forest_pool <- wrap_plots(
   wrap_elements(cindex_forest_pool) + wrap_elements(delta_forest_pool) +
    plot_layout(nrow = 2, heights = c(1, 0.8))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))

forest_pool <- wrap_plots(
  wrap_elements(forest_pool) + plot_spacer() + legend +
    plot_layout(nrow = 3, heights = c(1, -0.06, 0.1))
  )

showtext_auto(enable = T)
font_add("Arial", regular = "Arial.ttf")
ggsave(str_c(getwd(), "/results/Figure/Performance_comparison.pdf"), plot = forest_pool, width = 16, height = 16, unit = "in", dpi = 500, device = "pdf")
showtext_auto(enable = F)

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison.tiff"), plot = forest_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison.png"), plot = forest_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "png")

```

------------------------------------------------------------------------------------

# Supplementary Fig.6: Multiomics vs single omics

## PRS

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELPRSMet", "PANELPRSPro", "PANELPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANELPRS"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+PRS +MetScore", "+PRS +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(!model %in% c("PANEL"), glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model %in% c("PANEL", "+PRS"), "white", "#F39B7FCC")
  )

#### Plotting ####
f1_panel_prs <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_panel_prs <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_panel_prs <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_panel_prs <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_panel_prs <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_panel_prs <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_panel_prs <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_panel_prs <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_panel_prs <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```


## MetScore

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c("PANELMet", "PANELPRSMet", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANELMet"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("+MetScore", "+PRS +MetScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(!model %in% c("PANEL"), glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("*+MetScore*", "*+PRS +MetScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model %in% c("+MetScore"), "white", "#F39B7FCC")
  )

#### Plotting ####
f1_panel_metscore <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_panel_metscore <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_panel_metscore <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_panel_metscore <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_panel_metscore <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_panel_metscore <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_panel_metscore <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_panel_metscore <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_panel_metscore <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

## ProScore

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c("PANELPro", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANELPro"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("+ProScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(!model %in% c("PANEL"), glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("*+ProScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model %in% c("+ProScore"), "white", "#F39B7FCC")
  )

dat_sig <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0)

#### Plotting ####
f1_panel_proscore <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_panel_proscore <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_panel_proscore <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_panel_proscore <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_panel_proscore <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_panel_proscore <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_panel_proscore <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_panel_proscore <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_panel_proscore <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 4, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

## Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_panel_prs + f1_panel_metscore + f1_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_panel_prs + f2_panel_metscore + f2_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_panel_prs + f3_panel_metscore + f3_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_panel_prs + f4_panel_metscore + f4_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_panel_prs + f5_panel_metscore + f5_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
 f6_panel_prs + f6_panel_metscore + f6_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_panel_prs + f7_panel_metscore + f7_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_panel_prs + f8_panel_metscore + f8_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_panel_prs + f9_panel_metscore + f9_panel_proscore +
    plot_layout(nrow = 3, heights = c(1, 4/5, 4/5))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
delta_forest_pool <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    caption = "\u0394 C-index",
    theme = theme(
      plot.caption = element_markdown(size = 14, hjust = 0.5, vjust = 1, family = "Arial")
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

showtext_auto(enable = T)
font_add("Arial", regular = "Arial.ttf")
ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_single_omics.pdf"), plot = delta_forest_pool, width = 16, height = 7, unit = "in", dpi = 500, device = "pdf")
showtext_auto(enable = F)

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_single_omics.png"), plot = delta_forest_pool, width = 16, height = 7, unit = "in", dpi = 300, device = "png")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_single_omics.tiff"), plot = delta_forest_pool, width = 16, height = 7, unit = "in", dpi = 300, device = "tiff")

```

------------------------------------------------------------------------------------

# Supplementary Fig.7: C-index comparison -- biomarker

## C-index forest plot

### Biomarker only

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("NTproBNP", "NPPB", "Creatinine", "Albumin", "Gly", "GlycA")) %>%
  mutate(
    model = factor(model, levels = rev(c("NTproBNP", "NPPB", "Creatinine", "Albumin", "Gly", "GlycA")), labels = rev(c("NTproBNP", "NPPB", "Creatinine", "Albumin", "Gly", "GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(9)[6]

#### Plotting ####
f1_biomarker <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_biomarker <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_biomarker <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_biomarker <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_biomarker <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_biomarker <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_biomarker <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_biomarker <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_biomarker <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#8491B4CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#8491B4CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c("AgeSex", "AgeSexNTproBNP", "AgeSexNPPB", "AgeSexCreatinine", "AgeSexAlbumin", "AgeSexGly", "AgeSexGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+NT-proBNP", "+NPPB", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+NT-proBNP*", "*+NPPB*", "*+Creatinine*", "*+Albumin*", "*+Gly*", "*+GlycA*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(9)[6]

#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c("Clin", "ClinNTproBNP", "ClinNPPB", "ClinCreatinine", "ClinAlbumin", "ClinGly", "ClinGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+NT-proBNP", "+NPPB", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+NT-proBNP*", "*+NPPB*", "*+Creatinine*", "*+Albumin*", "*+Gly*", "*+GlycA*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(9)[6]

#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c("PANEL", "PANELNTproBNP", "PANELNPPB", "PANELCreatinine", "PANELAlbumin", "PANELGly", "PANELGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+NT-proBNP", "+NPPB", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+NT-proBNP*", "*+NPPB*", "*+Creatinine*", "*+Albumin*", "*+Gly*", "*+GlycA*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(9)[6]

#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_biomarker + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_biomarker + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_biomarker + f3_agesex + f3_clin + f3_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_biomarker + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_biomarker + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_biomarker + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_biomarker + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_biomarker + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_biomarker + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(5/6, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_biomarker <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.6, 8)))
) + 
  plot_annotation(
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_biomarker.tiff"), plot = cindex_forest_biomarker, width = 16, height = 9, unit = "in", dpi = 300, device = "tiff")

# ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_biomarker.pdf"), plot = cindex_forest_biomarker, width = 16, height = 9, unit = "in", dpi = 300, device = "pdf")

# ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_biomarker.png"), plot = cindex_forest_biomarker, width = 16, height = 9, unit = "in", dpi = 300, device = "png")


#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Biomarker
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("NTproBNP", "Creatinine", "Albumin", "Gly", "GlycA")) %>%
  mutate(
    model = factor(model, levels = rev(c("NTproBNP", "Creatinine", "Albumin", "Gly", "GlycA")), labels = rev(c("NTproBNP", "Creatinine", "Albumin", "Gly", "GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Biomarker"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c("AgeSex", "AgeSexNTproBNP", "AgeSexCreatinine", "AgeSexAlbumin", "AgeSexGly", "AgeSexGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+NT-proBNP", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c("Clin", "ClinNTproBNP", "ClinCreatinine", "ClinAlbumin", "ClinGly", "ClinGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+NT-proBNP", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c("PANEL", "PANELNTproBNP", "PANELGly", "PANELGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+NT-proBNP", "+Gly", "+GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_biomarker.tsv"))

```

## Delta C-index

### AgeSex-based

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c("AgeSex", "AgeSexNTproBNP", "AgeSexNPPB", "AgeSexCreatinine", "AgeSexAlbumin", "AgeSexGly", "AgeSexGlycA")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "AgeSex"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+NT-proBNP", "+NPPB", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+NT-proBNP*", "*+NPPB*", "*+Creatinine*", "*+Albumin*", "*+Gly*", "*+GlycA*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model == "AgeSex", "white", "#00A087CC")
  )


dat_age_sex <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0) %>% 
  mutate(
    delta_cindex_text = sprintf("%0.3f (%0.3f–%0.3f)", mean_delta_cindex, delta_cindex_2_5, delta_cindex_97_5)
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+NT-proBNP"),
    aes(
      x = model_italic, xend = model_italic,
      y = -0.00732, yend = 0.0565
    ),
    linewidth = 1, color = "#00A087CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+NT-proBNP"),
    aes(
      x = model_italic, xend = model_italic,
      y = -0.00732, yend = -0.039
    ),
    linewidth = 1, color = "#00A087CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

### Clinical-based

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c("Clin", "ClinNTproBNP", "ClinNPPB", "ClinCreatinine", "ClinAlbumin", "ClinGly", "ClinGlycA")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "Clin"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+NT-proBNP", "+NPPB", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+NT-proBNP*", "*+NPPB*", "*+Creatinine*", "*+Albumin*", "*+Gly*", "*+GlycA*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model == "Clin", "white", "#3C5488CC")
  )

dat_sig <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0)

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```

### PANEL-based

```{r}

#### Data preparation ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c("PANEL", "PANELNTproBNP", "PANELNPPB", "PANELCreatinine", "PANELAlbumin", "PANELGly", "PANELGlycA")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANEL"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+NT-proBNP", "+NPPB", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+NT-proBNP*", "*+NPPB*", "*+Creatinine*", "*+Albumin*", "*+Gly*", "*+GlycA*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = ifelse(model == "PANEL", "white", "#F39B7FCC")
  )

dat_sig <- dat %>%
  filter(delta_cindex_2_5 > 0 & delta_cindex_97_5 > 0)

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", "0.04", "0.08")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.11), breaks = c(0, 0.05, 0.1), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.09), breaks = c(0, 0.04, 0.08), labels = c("0", 0.04, 0.08)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.03, 0.15), breaks = c(0, 0.07, 0.14), labels = c("0", 0.07, 0.14)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.05, 0.17), breaks = c(-0.04, 0, 0.08, 0.16), labels = c(-0.04, "0", 0.08, 0.16)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.11), breaks = c(0, 0.05, 0.10), labels = c("0", 0.05, "0.10")) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.13), breaks = c(0, 0.06, 0.12), labels = c("0", 0.06, 0.12)) +
  coord_flip() + 
  annotate("text", x = 7, y = 0, size = 4, color = "black", label = "Baseline", hjust = -.1)

```


### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 3, heights = c(1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
delta_forest_biomarker <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.6, 8)))
) + 
  plot_annotation(
    caption = "\u0394 C-index",
    theme = theme(
      plot.caption = element_markdown(size = 14, hjust = 0.5, vjust = 1, family = "Arial")
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


#### Source data ####
cindex_raw <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

delta_cindex <- tibble()

# AgeSex
models <- c("AgeSex", "AgeSexNTproBNP", "AgeSexCreatinine", "AgeSexAlbumin", "AgeSexGly", "AgeSexGlycA")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+NT-proBNP", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

# Clin
models <- c("Clin", "ClinNTproBNP", "ClinCreatinine", "ClinAlbumin", "ClinGly", "ClinGlycA")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "Clin"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+NT-proBNP", "+Creatinine", "+Albumin", "+Gly", "+GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

# PANEL
models <- c("PANEL", "PANELNTproBNP", "PANELGly", "PANELGlycA")

dat <- cindex_raw %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  group_by(outcome, fold) %>% 
  mutate(delta_cindex = cindex - cindex[model == "PANEL"]) %>%
  ungroup() %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+NT-proBNP", "+Gly", "+GlycA"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

write_tsv(delta_cindex, file = str_c(getwd(), "/results/Table/Source data/delta_cindex_biomarker.tsv"))

```

## Pool

```{r}

# Legend
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & !model %in% c("PRSMet", "PRSMetPro", "PRSPro", "MetPro", "PRS", "Met", "Pro", "ASCVD", "SCORE2")) %>%
  mutate(
    group = case_when(
      model %in% c("NTproBNP", "NPPB", "Creatinine", "Albumin", "Gly", "GlycA") ~ "Individual biomarker",
      str_detect(model, "AgeSex") ~ "AgeSex-based",
      str_detect(model, "Clin") ~ "Clinical-based",
      str_detect(model, "PANEL") ~ "PANEL-based"
    ),
    group = factor(group, levels = c("Individual biomarker", "AgeSex-based", "Clinical-based", "PANEL-based")),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

legend <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex, colour = group)
  ) +
  geom_point(size = 2, alpha = 1) +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1) + 
  theme_light() +
  theme(
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.title = element_text(size = 12, hjust = .5)
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.4, 1)) +
  scale_colour_manual(
    values = c("#8491B4CC", "#00A087CC", "#3C5488CC", "#F39B7FCC")
  )

legend <- wrap_elements(ggpubr::get_legend(legend)) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


forest_biomarker <- wrap_plots(
   wrap_elements(cindex_forest_biomarker) + wrap_elements(delta_forest_biomarker) +
    plot_layout(nrow = 2, heights = c(1, 0.8))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))

forest_biomarker <- wrap_plots(
  wrap_elements(forest_biomarker) + plot_spacer() + legend +
    plot_layout(nrow = 3, heights = c(1, -0.06, 0.1))
  )

showtext_auto(enable = T)
font_add("Arial", regular = "Arial.ttf")
ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_biomarker.pdf"), plot = forest_biomarker, width = 14, height = 16, unit = "in", dpi = 500, device = "pdf")
showtext_auto(enable = F)

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_biomarker.tiff"), plot = forest_biomarker, width = 14, height = 16, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_biomarker.png"), plot = forest_biomarker, width = 14, height = 16, unit = "in", dpi = 300, device = "png")

```

------------------------------------------------------------------------------------

# C-index comparison -- Subgroup

## Age under 60

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_under_60/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.522, yend = 0.956
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.522, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "Met"),
    aes(
      x = model, xend = model,
      y = 0.644, yend = 0.885
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "Met"),
    aes(
      x = model, xend = model,
      y = 0.644, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_under_60/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_under_60/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_under_60/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_age_under_60 <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("a"), "  Aged less than 60")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_under_60/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_age_unedr_60.tsv"))


cindex_by_model_age_unedr_60 <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```


## Age over 60

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_over_60/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_over_60/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_over_60/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_over_60/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_age_over_60 <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("b"), "  Aged 60 and over")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Age_over_60/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_age_over_60.tsv"))


cindex_by_model_age_over_60 <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## Male

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Male/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Male/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Male/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Male/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_male <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("c"), "  Male")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Male/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_male.tsv"))


cindex_by_model_male <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## Female

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Female/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.516, yend = 0.804
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.516, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Female/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Female/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Female/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_female <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("d"), "  Female")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Female/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_female.tsv"))


cindex_by_model_female <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## White

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/White/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/White/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/White/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/White/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_white <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/White/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_white.tsv"))


cindex_by_model_white <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## Lipid-lowering therapy

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  # geom_segment(
  #   data = dat %>% filter(outcome == "PAD", model == "PRS"),
  #   aes(
  #     x = model, xend = model,
  #     y = 0.522, yend = 0.703
  #   ),
  #   linewidth = 1, color = "#E64B35CC"
  # ) +
  # geom_segment(
  #   data = dat %>% filter(outcome == "PAD", model == "PRS"),
  #   aes(
  #     x = model, xend = model,
  #     y = 0.522, yend = 0.39
  #   ),
  #   linewidth = 1, color = "#E64B35CC",
  #   arrow = arrow(length = unit(0.2, "cm"))
  # ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.495, yend = 0.742
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.495, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_lipid_lowering_therapy <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("a"), "  Lipid-lowering therapy")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_lipid_lowering_therapy.tsv"))


cindex_by_model_lipid_lowering_therapy <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## No lipid-lowering therapy

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.595, yend = 0.739
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.595, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_no_lipid_lowering_therapy <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("b"), "  No lipid-lowering therapy")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_lipid_lowering_therapy/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_no_lipid_lowering_therapy.tsv"))


cindex_by_model_no_lipid_lowering_therapy <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## Antihypertensive therapy

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "HF", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.431, yend = 0.512
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "HF", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.431, yend = 0.38
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "VA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.581, yend = 0.756
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "VA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.581, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.417, yend = 0.616
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.417, yend = 0.35
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "CVDeath", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.516, yend = 0.652
    ),
    linewidth = 1, color = "#E64B35CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "CVDeath", model == "PRS"),
    aes(
      x = model, xend = model,
      y = 0.516, yend = 0.41
    ),
    linewidth = 1, color = "#E64B35CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_antihypertensive_therapy <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("c"), "  Antihypertensive therapy")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/Antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_antihypertensive_therapy.tsv"))


cindex_by_model_antihypertensive_therapy <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```

## No antihypertensive therapy

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.37, 0.94), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.83), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.40, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_no_antihypertensive_therapy <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.4, 8)))
) + 
  plot_annotation(
    expression(paste(bold("d"), "  No antihypertensive therapy")),
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/No_antihypertensive_therapy/cindex_summary_df.csv"), show_col_types = F)

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(outcome != "Major adverse cardiovascular events" & model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+MetScore", "+ProScore", "+PRS +MetScore", "+PRS +ProScore", "+MetScore +ProScore", "+PRS +MetScore +ProScore"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_no_antihypertensive_therapy.tsv"))


cindex_by_model_no_antihypertensive_therapy <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex))

```


## Pool

### Supplementary Fig.8: By age and sex

```{r}

forest_by_age_sex_pool <- wrap_plots(
   wrap_elements(cindex_forest_age_under_60) +
     wrap_elements(cindex_forest_age_over_60) + 
     wrap_elements(cindex_forest_male) + 
     wrap_elements(cindex_forest_female) +
    plot_layout(ncol = 2, widths = c(1, 1, 1, 1))
   )

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_subgroup_by_age_sex.pdf"), plot = forest_by_age_sex_pool, width = 32, height = 18, unit = "in", dpi = 300, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_subgroup_by_age_sex.tiff"), plot = forest_by_age_sex_pool, width = 32, height = 18, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_subgroup_by_age_sex.png"), plot = forest_by_age_sex_pool, width = 32, height = 18, unit = "in", dpi = 300, device = "png")

```

### Supplementary Fig.9: By lipid-lowering and antihypertensive therapy

```{r}

forest_by_therapy_pool <- wrap_plots(
   wrap_elements(cindex_forest_lipid_lowering_therapy) +
     wrap_elements(cindex_forest_no_lipid_lowering_therapy) + 
     wrap_elements(cindex_forest_antihypertensive_therapy) + 
     wrap_elements(cindex_forest_no_antihypertensive_therapy) +
    plot_layout(ncol = 2, widths = c(1, 1, 1, 1))
   )

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_subgroup_by_therapy.pdf"), plot = forest_by_therapy_pool, width = 32, height = 18, unit = "in", dpi = 300, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_subgroup_by_therapy.tiff"), plot = forest_by_therapy_pool, width = 32, height = 18, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_subgroup_by_therapy.png"), plot = forest_by_therapy_pool, width = 32, height = 18, unit = "in", dpi = 300, device = "png")

```

------------------------------------------------------------------------------------

# Supplementary Fig.10: C-index comparison -- XGBoost

## C-index forest plot

### Single omics

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_summary_df.csv"), show_col_types = FALSE)

dat <- cindex %>%
  filter(model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#E64B35CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#E64B35CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_summary_df.csv"), show_col_types = FALSE)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+Met*", "*+Pro*", "*+PRS +Met*", "*+PRS +Pro*", "*+Met +Pro*", "*+PRS +Met +Pro*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+Pro"),
    aes(
      x = model, xend = model,
      y = 0.493, yend = 0.577
    ),
    linewidth = 1, color = "#00A087CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+Pro"),
    aes(
      x = model, xend = model,
      y = 0.493, yend = 0.37
    ),
    linewidth = 1, color = "#00A087CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#00A087CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#00A087CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```


### Clinical-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_summary_df.csv"), show_col_types = FALSE)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+Met*", "*+Pro*", "*+PRS +Met*", "*+PRS +Pro*", "*+Met +Pro*", "*+PRS +Met +Pro*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+PRS +Met +Pro"),
    aes(
      x = model, xend = model,
      y = 0.495, yend = 0.565
    ),
    linewidth = 1, color = "#3C5488CC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+PRS +Met +Pro"),
    aes(
      x = model, xend = model,
      y = 0.495, yend = 0.37
    ),
    linewidth = 1, color = "#3C5488CC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#3C5488CC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#3C5488CC") + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_summary_df.csv"), show_col_types = FALSE)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+Met*", "*+Pro*", "*+PRS +Met*", "*+PRS +Pro*", "*+Met +Pro*", "*+PRS +Met +Pro*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.45, 0.81), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.82), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.38, 0.86), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.49, 0.85), breaks = c(0.5, 0.6, 0.7, 0.8)) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.88), breaks = c(0.4, 0.6, 0.8)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.34, 0.92), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+Met +Pro"),
    aes(
      x = model, xend = model,
      y = 0.482, yend = 0.535
    ),
    linewidth = 1, color = "#F39B7FCC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+Met +Pro"),
    aes(
      x = model, xend = model,
      y = 0.482, yend = 0.37
    ),
    linewidth = 1, color = "#F39B7FCC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+PRS +Pro"),
    aes(
      x = model, xend = model,
      y = 0.493, yend = 0.560
    ),
    linewidth = 1, color = "#F39B7FCC"
  ) +
  geom_segment(
    data = dat %>% filter(outcome == "AAA", model == "+PRS +Pro"),
    aes(
      x = model, xend = model,
      y = 0.493, yend = 0.37
    ),
    linewidth = 1, color = "#F39B7FCC",
    arrow = arrow(length = unit(0.2, "cm"))
  ) +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.36, 1.02), breaks = c(0.4, 0.6, 0.8, 1.0)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.43, 0.78), breaks = c(0.5, 0.6, 0.7)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_cindex)
  ) +
  geom_point(size = 2, alpha = 1, color = "#F39B7FCC") +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1, color = "#F39B7FCC") + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 12, vjust = 0, angle = 00),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.39, 0.91), breaks = c(0.5, 0.7, 0.9)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
cindex_forest_pool <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.53, 8)))
) + 
  plot_annotation(
    caption = "C-index",
    theme = theme(
      plot.caption = element_text(size = 14, hjust = 0.5, vjust = 1)
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


#### Source data ####
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_summary_df.csv"))

cindex_all <- tibble()

# Single omics
dat <- cindex %>%
  filter(model %in% c("PRS", "Met", "Pro")) %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

cindex_all <- bind_rows(cindex_all, dat)

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

cindex_all <- bind_rows(cindex_all, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

cindex_all <- bind_rows(cindex_all, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

cindex_all <- bind_rows(cindex_all, dat)

cindex_by_model_xgboost <- cindex_all %>%
  group_by(predictor, model) %>%
  summarise(mean_cindex = mean(mean_cindex)) 

write_tsv(cindex_all, file = str_c(getwd(), "/results/Table/Source data/cindex_xgboost.tsv"))

```

## Delta C-index -- CPH vs XGBoost

### Single omics

```{r}

#### Data preparation ####
cindex_xgboost <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_results_df.csv"), show_col_types = F)

cindex_cph <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% c("PRS", "Met", "Pro")) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = "#E64B35CC"
  )


# color
pal_npg("nrc", alpha = 0.8)(5)[1]

#### Plotting ####
f1_omics <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f2_omics <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "Stroke"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip()

f3_omics <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "HF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip()

f4_omics <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AF"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.15, 0.30), labels = c("0", 0.15, "0.30")) +  
  coord_flip()

f5_omics <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.15, 0.31), breaks = c(-0.14, 0, 0.14, 0.28), labels = c(-0.14, "0", 0.14, 0.28)) +
  coord_flip()

f6_omics <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "PAD"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip()

f7_omics <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "AAA"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip()

f8_omics <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "VTE"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f9_omics <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = "CVDeath"
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.39), breaks = c(0, 0.12, 0.24, 0.36), labels = c("0", 0.12, 0.24, 0.36)) +
  coord_flip()

```

### AgeSex-based

```{r}

#### Data preparation ####
cindex_xgboost <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_results_df.csv"), show_col_types = F)

cindex_cph <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")


dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% models) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    model_italic = ifelse(model != "AgeSex", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("AgeSex", "*+PRS*", "*+Met*", "*+Pro*", "*+PRS +Met*", "*+PRS +Pro*", "*+Met +Pro*", "*+PRS +Met +Pro*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = "#00A087CC"
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[3]


#### Plotting ####
f1_agesex <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f2_agesex <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip() 

f3_agesex <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip()

f4_agesex <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.15, 0.30), labels = c("0", 0.15, "0.30")) +
  coord_flip() 

f5_agesex <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.15, 0.31), breaks = c(-0.14, 0, 0.14, 0.28), labels = c(-0.14, "0", 0.14, 0.28)) +
  coord_flip()

f6_agesex <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip() 

f7_agesex <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip() 

f8_agesex <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip() 

f9_agesex <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.39), breaks = c(0, 0.12, 0.24, 0.36), labels = c("0", 0.12, 0.24, 0.36)) +
  coord_flip()

```

### Clinical-based

```{r}

#### Data preparation ####
cindex_xgboost <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_results_df.csv"), show_col_types = F)

cindex_cph <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% models) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    model_italic = ifelse(model != "Clin", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("Clin", "*+PRS*", "*+Met*", "*+Pro*", "*+PRS +Met*", "*+PRS +Pro*", "*+Met +Pro*", "*+PRS +Met +Pro*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = "#3C5488CC"
  )


# color
pal_npg("nrc", alpha = 0.8)(5)[4]


#### Plotting ####
f1_clin <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    # axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f2_clin <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip()

f3_clin <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip() 

f4_clin <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.15, 0.30), labels = c("0", 0.15, "0.30")) +
  coord_flip()

f5_clin <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.15, 0.31), breaks = c(-0.14, 0, 0.14, 0.28), labels = c(-0.14, "0", 0.14, 0.28)) +
  coord_flip()

f6_clin <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip()

f7_clin <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip()

f8_clin <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f9_clin <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) +  
  theme_bw() +
  theme(
    # axis.text.x = element_text(size = 10, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_blank(),
    # axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.x = element_blank(),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    # panel.grid.major.y = element_line(color = "#DCDCDC", linetype = "solid"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.39), breaks = c(0, 0.12, 0.24, 0.36), labels = c("0", 0.12, 0.24, 0.36)) +
  coord_flip()

```

### PANEL-based

```{r}

#### Data preparation ####
cindex_xgboost <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_results_df.csv"), show_col_types = F)

cindex_cph <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% models) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    model_italic = ifelse(model != "PANEL", glue("*{model}*"), glue("{model}")),
    model_italic = factor(model_italic, levels = rev(c("PANEL", "*+PRS*", "*+Met*", "*+Pro*", "*+PRS +Met*", "*+PRS +Pro*", "*+Met +Pro*", "*+PRS +Met +Pro*"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    color = "#F39B7FCC"
  )

# color
pal_npg("nrc", alpha = 0.8)(5)[5]


#### Plotting ####
f1_panel <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_markdown(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f2_panel <- ggplot(
  dat %>% filter(outcome == "Stroke"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip()

f3_panel <- ggplot(
  dat %>% filter(outcome == "HF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.04, 0.32), breaks = c(0, 0.10, 0.20, 0.30), labels = c("0", "0.10", "0.20", "0.30")) +
  coord_flip()

f4_panel <- ggplot(
  dat %>% filter(outcome == "AF"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.32), breaks = c(0, 0.15, 0.30), labels = c("0", 0.15, "0.30")) +
  coord_flip()

f5_panel <- ggplot(
  dat %>% filter(outcome == "VA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.15, 0.31), breaks = c(-0.14, 0, 0.14, 0.28), labels = c(-0.14, "0", 0.14, 0.28)) +
  coord_flip()

f6_panel <- ggplot(
  dat %>% filter(outcome == "PAD"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip()

f7_panel <- ggplot(
  dat %>% filter(outcome == "AAA"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.17, 0.49), breaks = c(-0.15, 0, 0.15, 0.30, 0.45), labels = c(-0.15, "0", 0.15, "0.30", 0.45)) +
  coord_flip()

f8_panel <- ggplot(
  dat %>% filter(outcome == "VTE"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.01, 0.25), breaks = c(0, 0.12, 0.24), labels = c("0", 0.12, 0.24)) +
  coord_flip()

f9_panel <- ggplot(
  dat %>% filter(outcome == "CVDeath"), 
  aes(x = model_italic, y = mean_delta_cindex, color = color)
  ) +
  geom_hline(
    yintercept = 0,
    color = "#D3D3D3", linetype = "dashed", linewidth = 0.75
    ) + 
  geom_point(size = 3, alpha = 1, shape = 15) +
  geom_linerange(aes(ymin = delta_cindex_2_5, ymax = delta_cindex_97_5), linewidth = 1) + 
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 10, colour = "black", angle = 0),
    axis.text.y = element_blank(),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 12, vjust = 1, angle = 90),
    axis.ticks.y = element_blank(),
    panel.grid = element_blank(),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "none",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_colour_identity() +
  scale_y_continuous(expand = c(0, 0), limit = c(-0.02, 0.39), breaks = c(0, 0.12, 0.24, 0.36), labels = c("0", 0.12, 0.24, 0.36)) +
  coord_flip()

```

### Pool

```{r}

# Each column
col1 <- wrap_elements(
 f1_omics + f1_agesex + f1_clin + f1_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col2 <- wrap_elements(
  f2_omics + f2_agesex + f2_clin + f2_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col3 <- wrap_elements(
  f3_omics + f3_agesex + f3_clin + f3_panel + 
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col4 <- wrap_elements(
  f4_omics + f4_agesex + f4_clin + f4_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col5 <- wrap_elements(
  f5_omics + f5_agesex + f5_clin + f5_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col6 <- wrap_elements(
  f6_omics + f6_agesex + f6_clin + f6_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col7 <- wrap_elements(
  f7_omics + f7_agesex + f7_clin + f7_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col8 <- wrap_elements(
  f8_omics + f8_agesex + f8_clin + f8_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

col9 <- wrap_elements(
  f9_omics + f9_agesex + f9_clin + f9_panel +
    plot_layout(nrow = 4, heights = c(3/8, 1, 1, 1))
  ) + theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))

# Overall
delta_forest_pool <- wrap_elements(
  col1 + col2 + col3 + col4 + col5 + col6 + col7 + col8 + col9 +
    plot_layout(ncol = 9, widths = c(1, rep(0.53, 8)))
) + 
  plot_annotation(
    caption = "\u0394 C-index",
    theme = theme(
      plot.caption = element_markdown(size = 14, hjust = 0.5, vjust = 1, family = "Arial")
    )
  ) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


#### Source data ####
cindex_xgboost <- read_csv(str_c(getwd(), "/data/CindexComparison/All/xgboost_cindex_results_df.csv"), show_col_types = F)

cindex_cph <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_results_df.csv"), show_col_types = F)

delta_cindex <- tibble()

# Single omics
dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% c("PRS", "Met", "Pro")) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(model, levels = rev(c("PRS", "Met", "Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Omics"
  )

# AgeSex
models <- c(
  "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", 
  "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro")

dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% models) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("AgeSex", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "AgeSex"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

# Clin
models <- c(
  "Clin", "ClinPRS", "ClinMet", "ClinPro", 
  "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro")

dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% models) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("Clin", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "Clin"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

# PANEL
models <- c(
  "PANEL", "PANELPRS", "PANELMet", "PANELPro", 
  "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex_cph %>%
  left_join(cindex_xgboost, by = c("model", "outcome", "fold"), suffix = c("_cph", "_xgboost")) %>%
  filter(model %in% models) %>%
  mutate(delta_cindex = cindex_cph - cindex_xgboost) %>%
  group_by(outcome, model) %>%
  summarise(
    mean_delta_cindex = mean(delta_cindex, na.rm = TRUE),
    delta_cindex_2_5 = quantile(delta_cindex, 0.025, na.rm = TRUE),
    delta_cindex_97_5 = quantile(delta_cindex, 0.975, na.rm = TRUE)
  ) %>%
  ungroup() %>%
  mutate(
    model = factor(
      model, levels = rev(models),
      labels = rev(c("PANEL", "+PRS", "+Met", "+Pro", "+PRS +Met", "+PRS +Pro", "+Met +Pro", "+PRS +Met +Pro"))),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    predictor = "PANEL"
  )

delta_cindex <- bind_rows(delta_cindex, dat)

write_tsv(delta_cindex, file = str_c(getwd(), "/results/Table/Source data/delta_cindex_cph_xgboost.tsv"))

```

## Pool

```{r}

# Legend
cindex <- read_csv(str_c(getwd(), "/data/CindexComparison/All/cindex_summary_df.csv"), show_col_types = F)

models <- c("PRS", "Met", "Pro", "AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro", "Clin", "ClinPRS", "ClinMet", "ClinPro", "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro", "PANEL", "PANELPRS", "PANELMet", "PANELPro", "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro")

dat <- cindex %>%
  filter(model %in% models) %>%
  mutate(
    group = case_when(
      model %in% c("PRS", "Met", "Pro") ~ "Omics-based",
      str_detect(model, "AgeSex") ~ "AgeSex-based",
      str_detect(model, "Clin") ~ "Clinical-based",
      str_detect(model, "PANEL") ~ "PANEL-based"
    ),
    group = factor(group, levels = c("Omics-based", "AgeSex-based", "Clinical-based", "PANEL-based")),
    outcome = factor(
      outcome, 
      levels = c("Coronary artery diseases", "Stroke", "Heart failure", "Atrial fibrillation", "Ventricular arrhythmias", "Peripheral artery diseases", "Abdominal aortic aneurysm", "Vein thrombosis", "Cardiovascular death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      )
  )

legend <- ggplot(
  dat %>% filter(outcome == "CAD"), 
  aes(x = model, y = mean_cindex, colour = group)
  ) +
  geom_point(size = 2, alpha = 1) +
  geom_linerange(aes(ymin = cindex_2_5, ymax = cindex_97_5), linewidth = 1) + 
  theme_light() +
  theme(
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.title = element_text(size = 12, hjust = .5)
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  scale_y_continuous(expand = c(0, 0), limit = c(0.4, 1)) +
  scale_colour_manual(
    values = c("#E64B35CC", "#00A087CC", "#3C5488CC", "#F39B7FCC")
  )

legend <- wrap_elements(ggpubr::get_legend(legend)) + 
  theme(plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"))


forest_pool <- wrap_plots(
   wrap_elements(cindex_forest_pool) + wrap_elements(delta_forest_pool) +
    plot_layout(nrow = 2, heights = c(1, 1))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))

forest_pool <- wrap_plots(
  wrap_elements(forest_pool) + plot_spacer() + legend +
    plot_layout(nrow = 3, heights = c(1, -0.06, 0.1))
  )

showtext_auto(enable = T)
font_add("Arial", regular = "Arial.ttf")
ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_CPH_XGBoost.pdf"), plot = forest_pool, width = 16, height = 16, unit = "in", dpi = 500, device = "pdf")
showtext_auto(enable = F)

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_CPH_XGBoost.tiff"), plot = forest_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Performance_comparison_CPH_XGBoost.png"), plot = forest_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "png")

```

------------------------------------------------------------------------------------

# Calibration and decision curve analysis

## Calibration

### AgeSex-based

```{r}

#### Data preparation ####
folder_path <- str_c(getwd(), "/data/Pred")
file_list <- list.files(path = folder_path, pattern = "*.csv", full.names = TRUE)

model_mapping <- c(
  'ASCVD' = 'ASCVD',
  'SCORE2' = 'SCORE2',
  'Genomics (CoxPH)' = 'PRS',
  'Genomics_Metabolomics (CoxPH)' = 'PRSMet',
  'Genomics_Proteomics (CoxPH)' = 'PRSPro',
  'Genomics_Metabolomics_Proteomics (CoxPH)' = 'PRSMetPro',
  'Metabolomics (CoxPH)' = 'Met',
  'Metabolomics_Proteomics (CoxPH)' = 'MetPro',
  'Proteomics (CoxPH)' = 'Pro',
  'AgeSex (CoxPH)' = 'AgeSex',
  'AgeSex_Genomics (CoxPH)' = 'AgeSexPRS',
  'AgeSex_Metabolomics (CoxPH)' = 'AgeSexMet',
  'AgeSex_Proteomics (CoxPH)' = 'AgeSexPro',
  'AgeSex_Genomics_Metabolomics (CoxPH)' = 'AgeSexPRSMet',
  'AgeSex_Genomics_Proteomics (CoxPH)' = 'AgeSexPRSPro',
  'AgeSex_Metabolomics_Proteomics (CoxPH)' = 'AgeSexMetPro',
  'AgeSex_Genomics_Metabolomics_Proteomics (CoxPH)' = 'AgeSexPRSMetPro',
  'Clinical (CoxPH)' = 'Clin',
  'Clinical_Genomics (CoxPH)' = 'ClinPRS',
  'Clinical_Metabolomics (CoxPH)' = 'ClinMet',
  'Clinical_Proteomics (CoxPH)' = 'ClinPro',
  'Clinical_Genomics_Metabolomics (CoxPH)' = 'ClinPRSMet',
  'Clinical_Genomics_Proteomics (CoxPH)' = 'ClinPRSPro',
  'Clinical_Metabolomics_Proteomics (CoxPH)' = 'ClinMetPro',
  'Clinical_Genomics_Metabolomics_Proteomics (CoxPH)' = 'ClinPRSMetPro',
  'PANEL (CoxPH)' = 'PANEL',
  'PANEL_Genomics (CoxPH)' = 'PANELPRS',
  'PANEL_Metabolomics (CoxPH)' = 'PANELMet',
  'PANEL_Proteomics (CoxPH)' = 'PANELPro',
  'PANEL_Genomics_Metabolomics (CoxPH)' = 'PANELPRSMet',
  'PANEL_Genomics_Proteomics (CoxPH)' = 'PANELPRSPro',
  'PANEL_Metabolomics_Proteomics (CoxPH)' = 'PANELMetPro',
  'PANEL_Genomics_Metabolomics_Proteomics (CoxPH)' = 'PANELPRSMetPro'
)

combined_data <- file_list %>%
  map_df(~ {
    file_name <- basename(.x) 
    model_name <- str_remove(file_name, "_test_pred.csv")
    model <- model_mapping[model_name]
    
    read_csv(.x, show_col_types = FALSE) %>%
      mutate(model = model) 
  })

dat <- combined_data %>% 
  filter(str_detect(model, "AgeSex")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  mutate(
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      model == "AgeSex" ~ "AgeSex",
      str_detect(model, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(model, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(model, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(model, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(model, "PRS$") ~ "+PRS",
      str_detect(model, "Met$") ~ "+MetScore",
      str_detect(model, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(group != "AgeSex", glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  group_by(outcome, model) %>%
  mutate(
    risk_group = ntile(predicted_prob15, 10),
    event_15 = case_when(
      event == 0 ~ 0,
      event == 1 & duration > 15 ~ 0,
      event == 1 & duration <= 15 ~ 1
    )
  ) %>%
  ungroup() %>%
  group_by(outcome, model, risk_group) %>% 
  mutate(
    mean_risk = mean(predicted_prob15), 
    mean_events = mean(event_15)
    ) %>%
  slice(1) %>%
  ungroup() %>%
  select(outcome, group_italic, mean_risk, mean_events)


#### Plotting ####
cal_plot_agesex <- ggplot(
  dat, 
  aes(x = mean_risk*100, y = mean_events*100, color = group_italic)
  ) + 
  geom_abline(intercept = 0, slope = 1, color = "gray", alpha = 1, linewidth = 0.75, linetype = "dashed") +
  geom_line(linewidth = 0.75) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = pal_npg("nrc", alpha = 0.8)(8)
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(xlim = c(0, NA), ylim = c(0, NA)) +
  # facet_wrap(~ outcome, scales = "free") +
  facet_wrap_equal(~ outcome, scales = "free") + 
  labs(
    x = "Predicted Risk (%)", 
    y = "Observed event rate (%)", 
    color = "")

```

### Clinical-based

```{r}

#### Data preparation ####
dat <- combined_data %>% 
  filter(str_detect(model, "Clin")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  mutate(
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      model == "Clin" ~ "Clin",
      str_detect(model, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(model, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(model, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(model, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(model, "PRS$") ~ "+PRS",
      str_detect(model, "Met$") ~ "+MetScore",
      str_detect(model, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(group != "Clin", glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  group_by(outcome, model) %>%
  mutate(
    risk_group = ntile(predicted_prob15, 10),
    event_15 = case_when(
      event == 0 ~ 0,
      event == 1 & duration > 15 ~ 0,
      event == 1 & duration <= 15 ~ 1
    )
  ) %>%
  ungroup() %>%
  group_by(outcome, model, risk_group) %>% 
  mutate(
    mean_risk = mean(predicted_prob15), 
    mean_events = mean(event_15),
    num_events = sum(event_15),
    num_people = n()
    ) %>%
  slice(1) %>%
  ungroup() %>%
  select(outcome, group_italic, mean_risk, mean_events, num_events, num_people)


#### Plotting ####
cal_plot_clin <- ggplot(
  dat, 
  aes(x = mean_risk*100, y = mean_events*100, color = group_italic)
  ) + 
  geom_abline(intercept = 0, slope = 1, color = "gray", alpha = 1, linewidth = 0.75, linetype = "dashed") +
  geom_line(linewidth = 0.75) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = pal_npg("nrc", alpha = 0.8)(8)
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +  
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(xlim = c(0, NA), ylim = c(0, NA)) +
  # facet_wrap(~ outcome, scales = "free") +
  facet_wrap_equal(~ outcome, scales = "free") +
  labs(
    x = "Predicted Risk (%)", 
    y = "Observed event rate (%)", 
    color = "")

```

### PANEL-based

```{r}

#### Data preparation ####
dat <- combined_data %>% 
  filter(str_detect(model, "PANEL")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  mutate(
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      model == "PANEL" ~ "PANEL",
      str_detect(model, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(model, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(model, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(model, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(model, "PRS$") ~ "+PRS",
      str_detect(model, "Met$") ~ "+MetScore",
      str_detect(model, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(group != "PANEL", glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  group_by(outcome, model) %>%
  mutate(
    risk_group = ntile(predicted_prob15, 10),
    event_15 = case_when(
      event == 0 ~ 0,
      event == 1 & duration > 15 ~ 0,
      event == 1 & duration <= 15 ~ 1
    )
  ) %>%
  ungroup() %>%
  group_by(outcome, model, risk_group) %>% 
  mutate(
    mean_risk = mean(predicted_prob15), 
    mean_events = mean(event_15)
    ) %>%
  slice(1) %>%
  ungroup() %>%
  select(outcome, group_italic, mean_risk, mean_events)


#### Plotting ####
cal_plot_panel <- ggplot(
  dat, 
  aes(x = mean_risk*100, y = mean_events*100, color = group_italic)
  ) + 
  geom_abline(intercept = 0, slope = 1, color = "gray", alpha = 1, linewidth = 0.75, linetype = "dashed") +
  geom_line(linewidth = 0.75) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = pal_npg("nrc", alpha = 0.8)(8)
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +  
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(xlim = c(0, NA), ylim = c(0, NA)) +
  # facet_wrap(~ outcome, scales = "free") +
  facet_wrap_equal(~ outcome, scales = "free") +
  labs(
    x = "Predicted Risk (%)", 
    y = "Observed event rate (%)", 
    color = "")

```

#### Example for Figure 1

```{r}

#### Data preparation ####
dat <- combined_data %>% 
  filter(str_detect(model, "PANEL")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  mutate(
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      model == "PANEL" ~ "PANEL",
      str_detect(model, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(model, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(model, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(model, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(model, "PRS$") ~ "+PRS",
      str_detect(model, "Met$") ~ "+MetScore",
      str_detect(model, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(group != "PANEL", glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  group_by(outcome, model) %>%
  mutate(
    risk_group = ntile(predicted_prob15, 10),
    event_15 = case_when(
      event == 0 ~ 0,
      event == 1 & duration > 15 ~ 0,
      event == 1 & duration <= 15 ~ 1
    )
  ) %>%
  ungroup() %>%
  group_by(outcome, model, risk_group) %>% 
  mutate(
    mean_risk = mean(predicted_prob15), 
    mean_events = mean(event_15)
    ) %>%
  slice(1) %>%
  ungroup() %>%
  select(outcome, group_italic, mean_risk, mean_events) %>%
  filter(outcome == "CAD")


#### Plotting ####
example <- ggplot(
  dat, 
  aes(x = mean_risk*100, y = mean_events*100, color = group_italic)
  ) + 
  geom_abline(intercept = 0, slope = 1, color = "gray", alpha = 1, linewidth = 0.75, linetype = "dashed") +
  geom_line(linewidth = 0.75) +
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0.5, r = 0.5, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = pal_npg("nrc", alpha = 0.8)(8)
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +  
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(xlim = c(0, NA), ylim = c(0, NA)) +
  labs(
    x = "Predicted Risk (%)", 
    y = "Observed event rate (%)", 
    color = "")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_example.tiff"), plot = example, width = 4, height = 3, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_example.png"), plot = example, width = 4, height = 3, unit = "in", dpi = 300, device = "png")

```


## Decision curve analysis

### AgeSex-based

```{r}

#### Data preparation ####
folder_path <- str_c(getwd(), "/data/Pred")
file_list <- list.files(path = folder_path, pattern = "*.csv", full.names = TRUE)

model_mapping <- c(
  'ASCVD' = 'ASCVD',
  'SCORE2' = 'SCORE2',
  'Genomics (CoxPH)' = 'PRS',
  'Genomics_Metabolomics (CoxPH)' = 'PRSMet',
  'Genomics_Proteomics (CoxPH)' = 'PRSPro',
  'Genomics_Metabolomics_Proteomics (CoxPH)' = 'PRSMetPro',
  'Metabolomics (CoxPH)' = 'Met',
  'Metabolomics_Proteomics (CoxPH)' = 'MetPro',
  'Proteomics (CoxPH)' = 'Pro',
  'AgeSex (CoxPH)' = 'AgeSex',
  'AgeSex_Genomics (CoxPH)' = 'AgeSexPRS',
  'AgeSex_Metabolomics (CoxPH)' = 'AgeSexMet',
  'AgeSex_Proteomics (CoxPH)' = 'AgeSexPro',
  'AgeSex_Genomics_Metabolomics (CoxPH)' = 'AgeSexPRSMet',
  'AgeSex_Genomics_Proteomics (CoxPH)' = 'AgeSexPRSPro',
  'AgeSex_Metabolomics_Proteomics (CoxPH)' = 'AgeSexMetPro',
  'AgeSex_Genomics_Metabolomics_Proteomics (CoxPH)' = 'AgeSexPRSMetPro',
  'Clinical (CoxPH)' = 'Clin',
  'Clinical_Genomics (CoxPH)' = 'ClinPRS',
  'Clinical_Metabolomics (CoxPH)' = 'ClinMet',
  'Clinical_Proteomics (CoxPH)' = 'ClinPro',
  'Clinical_Genomics_Metabolomics (CoxPH)' = 'ClinPRSMet',
  'Clinical_Genomics_Proteomics (CoxPH)' = 'ClinPRSPro',
  'Clinical_Metabolomics_Proteomics (CoxPH)' = 'ClinMetPro',
  'Clinical_Genomics_Metabolomics_Proteomics (CoxPH)' = 'ClinPRSMetPro',
  'PANEL (CoxPH)' = 'PANEL',
  'PANEL_Genomics (CoxPH)' = 'PANELPRS',
  'PANEL_Metabolomics (CoxPH)' = 'PANELMet',
  'PANEL_Proteomics (CoxPH)' = 'PANELPro',
  'PANEL_Genomics_Metabolomics (CoxPH)' = 'PANELPRSMet',
  'PANEL_Genomics_Proteomics (CoxPH)' = 'PANELPRSPro',
  'PANEL_Metabolomics_Proteomics (CoxPH)' = 'PANELMetPro',
  'PANEL_Genomics_Metabolomics_Proteomics (CoxPH)' = 'PANELPRSMetPro'
)

combined_data <- file_list %>%
  map_df(~ {
    file_name <- basename(.x) 
    model_name <- str_remove(file_name, "_test_pred.csv")
    model <- model_mapping[model_name] 
    
    read_csv(.x, show_col_types = FALSE) %>%
      mutate(model = model) 
  })


dat_wide <- combined_data %>% 
  filter(str_detect(model, "AgeSex")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  pivot_wider(names_from = model, values_from = predicted_prob15)

# Example
# dca(
#   Surv(duration, event) ~ AgeSex,
#   data = dat_wide %>% filter(outcome == "cad"),
#   time = 15,
#   thresholds = seq(0, 0.5, by = 0.1)
# ) %>%
# standardized_net_benefit() %>% # Standardize the net benefit
# plot(smooth = F, show_ggplot_code = TRUE)

tic()
dat_agesex <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for(i in 1:9){
  dca <- dca(
    Surv(duration, event) ~ AgeSex + AgeSexPRS + AgeSexMet + AgeSexPro + AgeSexPRSMet + AgeSexPRSPro + AgeSexMetPro + AgeSexPRSMetPro, 
    data = dat_wide %>% filter(outcome == outcomes[i]),
    time = 15,
    thresholds = seq(0, 0.5, by = 0.001)
    ) %>%
    standardized_net_benefit()
  
  dat_agesex <- bind_rows(dat_agesex, dca$dca %>% mutate(outcome = outcomes[i]))
  
  rm(dca)
}
toc() # 400 s


# Calculate the difference in the area under the net benefit curve
diff_nb_agesex <- dat_agesex %>%
  filter(
    variable %in% c("AgeSex", "AgeSexPRS", "AgeSexMet", "AgeSexPro", "AgeSexPRSMet", "AgeSexPRSPro", "AgeSexMetPro", "AgeSexPRSMetPro") & 
      !is.na(standardized_net_benefit)) %>%
  arrange(variable, outcome, threshold) %>%
  group_by(outcome, variable) %>%
  summarise(anbc = trapz(threshold*100, standardized_net_benefit*100), .groups = "drop") %>%
  group_by(outcome) %>%
  mutate(diff_anbc = anbc - anbc[variable == "AgeSex"]) %>%
  ungroup()

# Define threshold ranges for different outcomes
threshold_ranges <- tibble(
  outcome = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
  xmin = c(0, 0, 0, 0, 0, 0, 0, 0, 0),
  xmax = c(0.3, 0.10, 0.25, 0.35, 0.028, 0.075, 0.02, 0.10, 0.2)
  )

dat <- dat_agesex %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  left_join(diff_nb_agesex, by = c("outcome", "variable")) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "AgeSex" ~ "AgeSex",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "AgeSex"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic, diff_anbc)

dat_label <- dat %>% 
  filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None", "AgeSex")) %>%
  mutate(diff_anbc = sprintf("%0.2f", diff_anbc)) %>%
  group_by(outcome, group_italic) %>%
  slice_sample(n = 1) %>%
  ungroup() %>%
  mutate(title = "A-NBC difference:")

# color
# pal_npg("nrc", alpha = 0.8)(8)


#### Plotting ####
dca_plot_agesex <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold*100, y = standardized_net_benefit*100, color = group_italic)) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat All"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat None"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +  
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(ylim = c(-0.05*100, 1.02*100)) +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Threshold probability (%)", 
    y = "Standardized net benefit (%)", 
    color = ""
    ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS*"),
    mapping = aes(x = Inf, y = 100, label = title),
    inherit.aes = FALSE,
    size = 4, hjust = 1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS*"),
    mapping = aes(x = Inf, y = 90, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+MetScore*"),
    mapping = aes(x = Inf, y = 80, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+ProScore*"),
    mapping = aes(x = Inf, y = 70, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +MetScore*"),
    mapping = aes(x = Inf, y = 60, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +ProScore*"),
    mapping = aes(x = Inf, y = 50, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+MetScore +ProScore*"),
    mapping = aes(x = Inf, y = 40, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +MetScore +ProScore*"),
    mapping = aes(x = Inf, y = 30, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  )

```


### Clinical-based


```{r}

#### Data preparation ####
dat_wide <- combined_data %>% 
  filter(str_detect(model, "Clin")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  pivot_wider(names_from = model, values_from = predicted_prob15)

tic()
dat_clin <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for(i in 1:9){
  dca <- dca(
    Surv(duration, event) ~ Clin + ClinPRS + ClinMet + ClinPro + ClinPRSMet + ClinPRSPro + ClinMetPro + ClinPRSMetPro, 
    data = dat_wide %>% filter(outcome == outcomes[i]),
    time = 15,
    thresholds = seq(0, 0.5, by = 0.001)
    ) %>%
    standardized_net_benefit()
  
  dat_clin <- bind_rows(dat_clin, dca$dca %>% mutate(outcome = outcomes[i]))
  
  rm(dca)
}
toc() # 400 s


# Calculate the difference in the area under the net benefit curve
diff_nb_clin <- dat_clin %>%
  filter(
    variable %in% c("Clin", "ClinPRS", "ClinMet", "ClinPro", "ClinPRSMet", "ClinPRSPro", "ClinMetPro", "ClinPRSMetPro") & 
      !is.na(standardized_net_benefit)) %>%
  arrange(variable, outcome, threshold) %>%
  group_by(outcome, variable) %>%
  summarise(anbc = trapz(threshold*100, standardized_net_benefit*100), .groups = "drop") %>%
  group_by(outcome) %>%
  mutate(diff_anbc = anbc - anbc[variable == "Clin"]) %>%
  ungroup()

# Define threshold ranges for different outcomes
threshold_ranges <- tibble(
  outcome = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
  xmin = c(0, 0, 0, 0, 0, 0, 0, 0, 0),
  xmax = c(0.3, 0.10, 0.25, 0.4, 0.04, 0.12, 0.026, 0.10, 0.25)
  )

dat <- dat_clin %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  left_join(diff_nb_clin, by = c("outcome", "variable")) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "Clin" ~ "Clin",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "Clin"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic, diff_anbc)

dat_label <- dat %>% 
  filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None", "Clin")) %>%
  mutate(diff_anbc = sprintf("%0.2f", diff_anbc)) %>%
  group_by(outcome, group_italic) %>%
  slice_sample(n = 1) %>%
  ungroup() %>%
  mutate(title = "A-NBC difference:")

# color
# pal_npg("nrc", alpha = 0.8)(8)


#### Plotting ####
dca_plot_clin <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold*100, y = standardized_net_benefit*100, color = group_italic)) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat All"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat None"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(ylim = c(-0.05*100, 1.02*100)) +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Threshold probability (%)", 
    y = "Standardized net benefit (%)", 
    color = ""
    ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS*"),
    mapping = aes(x = Inf, y = 100, label = title),
    inherit.aes = FALSE,
    size = 4, hjust = 1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS*"),
    mapping = aes(x = Inf, y = 90, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+MetScore*"),
    mapping = aes(x = Inf, y = 80, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+ProScore*"),
    mapping = aes(x = Inf, y = 70, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +MetScore*"),
    mapping = aes(x = Inf, y = 60, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +ProScore*"),
    mapping = aes(x = Inf, y = 50, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+MetScore +ProScore*"),
    mapping = aes(x = Inf, y = 40, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +MetScore +ProScore*"),
    mapping = aes(x = Inf, y = 30, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  )

```

### PANEL-based

```{r}

#### Data preparation ####
dat_wide <- combined_data %>% 
  filter(str_detect(model, "PANEL")) %>%
  select(eid, outcome, fold, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  pivot_wider(names_from = model, values_from = predicted_prob15)

tic()
dat_panel <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for(i in 1:9){
  dca <- dca(
    Surv(duration, event) ~ PANEL + PANELPRS + PANELMet + PANELPro + PANELPRSMet + PANELPRSPro + PANELMetPro + PANELPRSMetPro, 
    data = dat_wide %>% filter(outcome == outcomes[i]),
    time = 15,
    thresholds = seq(0, 0.5, by = 0.001)
    ) %>%
    standardized_net_benefit()
  
  dat_panel <- bind_rows(dat_panel, dca$dca %>% mutate(outcome = outcomes[i]))
  
  rm(dca)
}
toc() # 400 s


# Calculate the difference in the area under the net benefit curve
diff_nb_panel <- dat_panel %>%
  filter(
    variable %in% c("PANEL", "PANELPRS", "PANELMet", "PANELPro", "PANELPRSMet", "PANELPRSPro", "PANELMetPro", "PANELPRSMetPro") & 
      !is.na(standardized_net_benefit)) %>%
  arrange(variable, outcome, threshold) %>%
  group_by(outcome, variable) %>%
  summarise(anbc = trapz(threshold*100, standardized_net_benefit*100), .groups = "drop") %>%
  # summarise(anbc = trapz(threshold, standardized_net_benefit), .groups = "drop") %>%
  group_by(outcome) %>%
  mutate(diff_anbc = anbc - anbc[variable == "PANEL"]) %>%
  ungroup()

# Define threshold ranges for different outcomes
threshold_ranges <- tibble(
  outcome = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
  xmin = c(0, 0, 0, 0, 0, 0, 0, 0, 0),
  xmax = c(0.3, 0.07, 0.25, 0.4, 0.04, 0.15, 0.04, 0.10, 0.20)
  )

dat <- dat_panel %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  left_join(diff_nb_panel, by = c("outcome", "variable")) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "PANEL" ~ "PANEL",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "PANEL"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic, diff_anbc)

dat_label <- dat %>% 
  filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None", "PANEL")) %>%
  mutate(diff_anbc = sprintf("%0.2f", diff_anbc)) %>%
  group_by(outcome, group_italic) %>%
  slice_sample(n = 1) %>%
  ungroup() %>%
  mutate(title = "A-NBC difference:")

# color
# pal_npg("nrc", alpha = 0.8)(8)


#### Plotting ####
dca_plot_panel <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold*100, y = standardized_net_benefit*100, color = group_italic)) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat All"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat None"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +  
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(ylim = c(-0.05*100, 1.02*100)) +
  facet_wrap(~ outcome, scales = "free") +
  labs(
    x = "Threshold probability (%)", 
    y = "Standardized net benefit (%)", 
    color = ""
    ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS*"),
    mapping = aes(x = Inf, y = 100, label = title),
    inherit.aes = FALSE,
    size = 4, hjust = 1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS*"),
    mapping = aes(x = Inf, y = 90, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+MetScore*"),
    mapping = aes(x = Inf, y = 80, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+ProScore*"),
    mapping = aes(x = Inf, y = 70, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +MetScore*"),
    mapping = aes(x = Inf, y = 60, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +ProScore*"),
    mapping = aes(x = Inf, y = 50, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+MetScore +ProScore*"),
    mapping = aes(x = Inf, y = 40, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  ) +
  geom_text(
    data = dat_label %>% filter(group_italic == "*+PRS +MetScore +ProScore*"),
    mapping = aes(x = Inf, y = 30, label = diff_anbc),
    size = 4, hjust = 1.1, vjust = 0.8
  )
  

```

#### Example for Figure 1

```{r}

#### Data preparation ####
dat_wide <- combined_data %>% 
  filter(str_detect(model, "PANEL")) %>%
  select(eid, outcome, model, predicted_prob15, event, duration) %>% # Predicted 15-year probabilty
  pivot_wider(names_from = model, values_from = predicted_prob15)

tic()
dat_panel <- tibble()
outcomes <- c("cad")
for(i in 1){
  dca <- dca(
    Surv(duration, event) ~ PANEL + PANELPRS + PANELMet + PANELPro + PANELPRSMet + PANELPRSPro + PANELMetPro + PANELPRSMetPro, 
    data = dat_wide %>% filter(outcome == outcomes[i]),
    time = 15,
    thresholds = seq(0, 0.5, by = 0.001)
    ) %>%
    standardized_net_benefit()
  
  dat_panel <- bind_rows(dat_panel, dca$dca %>% mutate(outcome = outcomes[i]))
  
  rm(dca)
}
toc() # 400 s

# Define threshold ranges for different outcomes
threshold_ranges <- tibble(
  outcome = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
  xmin = c(0, 0, 0, 0, 0, 0, 0, 0, 0),
  xmax = c(0.3, 0.07, 0.25, 0.4, 0.03, 0.08, 0.015, 0.08, 0.20)
  )

dat <- dat_panel %>%
  filter(outcome == "cad") %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad"),
      labels = c("CAD")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "PANEL" ~ "PANEL",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "PANEL"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic)


#### Plotting ####
example <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold*100, y = standardized_net_benefit*100, color = group_italic)) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat All"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(
    data = dat %>% filter(!is.na(standardized_net_benefit) & group_italic == "Treat None"),
    aes(x = threshold*100, y = standardized_net_benefit*100), 
    color = "gray", linetype = "dashed", linewidth = 0.75
  ) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "None",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0.5, r = 0.5, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  scale_x_continuous(labels = number_format(accuracy = 0.1)) +
  scale_y_continuous(labels = number_format(accuracy = 0.1)) +
  coord_cartesian(ylim = c(-0.05*100, 1.02*100)) +
  labs(
    x = "Threshold probability (%)", 
    y = "Standardized net benefit (%)", 
    color = ""
    )

ggsave(str_c(getwd(), "/results/Figure/Model_decision_curve_example.tiff"), plot = example, width = 4, height = 3, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Model_decision_curve_example.png"), plot = example, width = 4, height = 3, unit = "in", dpi = 300, device = "png")

```


## Pool

### Supplementary Fig.10: AgeSex-based

```{r}

# Legend
dat <- dat_agesex %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "AgeSex" ~ "AgeSex",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "AgeSex"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "AgeSex", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic)

legend <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold, y = standardized_net_benefit, color = group_italic)) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "right",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  coord_cartesian(ylim = c(-0.02, 1.02)) +
  facet_wrap(~ outcome, scales = "free") +
  labs(color = "")


#### Pool ####
cal_dca_pool <- wrap_plots(
   cal_plot_agesex + dca_plot_agesex +
    plot_layout(nrow = 2, heights = c(1, 1))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))

cal_dca_pool <- wrap_plots(
  wrap_elements(cal_dca_pool) + plot_spacer() + wrap_elements(ggpubr::get_legend(legend)) + plot_spacer() + 
    plot_layout(ncol = 4, widths = c(1, 0.05, 0.1, 0.05))
  )

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_AgeSex.pdf"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_AgeSex.tiff"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_AgeSex.png"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "png")

```

### Supplementary Fig.11: Clinical-based

```{r}

# Legend
dat <- dat_clin %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "Clin" ~ "Clin",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "Clin"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "Clin", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic)

legend <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold, y = standardized_net_benefit, color = group_italic)) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "right",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  coord_cartesian(ylim = c(-0.02, 1.02)) +
  facet_wrap(~ outcome, scales = "free") +
  labs(color = "")


#### Pool ####
cal_dca_pool <- wrap_plots(
   cal_plot_clin + dca_plot_clin +
    plot_layout(nrow = 2, heights = c(1, 1))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))

cal_dca_pool <- wrap_plots(
  wrap_elements(cal_dca_pool) + plot_spacer() + wrap_elements(ggpubr::get_legend(legend)) + plot_spacer() + 
    plot_layout(ncol = 4, widths = c(1, 0.05, 0.1, 0.05))
  )

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_Clin.pdf"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_Clin.tiff"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_Clin.png"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "png")

```

### Fig.4: PANEL-based

```{r}

# Legend
dat <- dat_panel %>%
  inner_join(threshold_ranges, by = "outcome") %>%
  filter(threshold >= xmin & threshold <= xmax) %>%
  select(-xmin, -xmax) %>%
  mutate(
    outcome = factor(
      outcome,
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
      ),
    group = case_when(
      label == "Treat All" ~ "Treat All",
      label == "Treat None" ~ "Treat None",
      label == "PANEL" ~ "PANEL",
      str_detect(label, "PRSMetPro$") ~ "+PRS +MetScore +ProScore",
      str_detect(label, "PRSMet$") ~ "+PRS +MetScore",
      str_detect(label, "PRSPro$") ~ "+PRS +ProScore",
      str_detect(label, "MetPro$") ~ "+MetScore +ProScore",
      str_detect(label, "PRS$") ~ "+PRS",
      str_detect(label, "Met$") ~ "+MetScore",
      str_detect(label, "Pro$") ~ "+ProScore"
    ),
    group_italic = ifelse(!group %in% c("Treat All", "Treat None", "PANEL"), glue("*{group}*"), glue("{group}")),
    group_italic = factor(group_italic, levels = c("Treat All", "Treat None", "PANEL", "*+PRS*", "*+MetScore*", "*+ProScore*", "*+PRS +MetScore*", "*+PRS +ProScore*", "*+MetScore +ProScore*", "*+PRS +MetScore +ProScore*"))
  ) %>%
  select(threshold, standardized_net_benefit, outcome, group_italic)

legend <- dat %>% filter(!is.na(standardized_net_benefit) & !group_italic %in% c("Treat All", "Treat None")) %>%
  ggplot(aes(x = threshold, y = standardized_net_benefit, color = group_italic)) +
  geom_line(linewidth = 0.75) + 
  theme_classic() +
  theme(
    axis.text.x = element_text(size = 12, colour = "black", angle = 0),
    axis.text.y = element_text(size = 12, colour = "black"),
    axis.title.x = element_text(size = 12, colour = "black", angle = 0, vjust = -1),
    axis.title.y = element_text(size = 12, colour = "black", vjust = 2),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_markdown(size = 12),
    legend.position = "right",
    plot.title = element_text(size = 12, hjust = .5),
    plot.margin = unit(c(t = 0, r = 0, b = 0.5, l = 0.5), "cm"),
    strip.background = element_blank(),
    strip.text = element_text(size = 12),
    panel.spacing = unit(1, "lines")
  ) +
  scale_colour_manual(
    values = c("#E64B35CC", "#4DBBD5CC", "#00A087CC", "#3C5488CC", "#F39B7FCC", "#8491B4CC", "#91D1C2CC", "#DC0000CC")
  ) + 
  coord_cartesian(ylim = c(-0.02, 1.02)) +
  facet_wrap(~ outcome, scales = "free") +
  labs(color = "")


#### Pool ####
cal_dca_pool <- wrap_plots(
   cal_plot_panel + dca_plot_panel +
    plot_layout(nrow = 2, heights = c(1, 1))
   ) &
  plot_annotation(tag_levels = "a") &
  theme(plot.tag = element_text(face = "bold"))

cal_dca_pool <- wrap_plots(
  wrap_elements(cal_dca_pool) + plot_spacer() + wrap_elements(ggpubr::get_legend(legend)) + plot_spacer() + 
    plot_layout(ncol = 4, widths = c(1, 0.055, 0.1, 0.05))
  )

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_PANEL.pdf"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_PANEL.tiff"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "tiff")

ggsave(str_c(getwd(), "/results/Figure/Model_calibration_decision_curve_PANEL.png"), plot = cal_dca_pool, width = 13, height = 15, unit = "in", dpi = 300, device = "png")

```


------------------------------------------------------------------------------------

# Supplementary Fig.12-13: SHAP importance

## Mean absolute SHAP value heatmap

### Top-ranked biomarkers

#### Metabolomics

```{r}

#### Data preparation ####
folder_path <- "your path/data/SHAP/Metabolomics"

tic()
mean_abs_shap_met <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for (i in 1:9) {
  shap_data <- read_shap_data(folder_path, outcomes[i])
  
  # Compute the mean absolute SHAP value
  mean_abs_shap <- bind_rows(shap_data) %>%
    summarise(across(-c(fold, sample), ~ mean(abs(.))))
  
  # Convert the results into the required format
  reshaped_data <- tibble(
    biomarker = names(mean_abs_shap),
    outcome = outcomes[i],
    shap = as.numeric(mean_abs_shap[1, ])
  )

  # Merge into the final dataframe
  mean_abs_shap_pro <- bind_rows(mean_abs_shap_pro, reshaped_data)
  
  rm(shap_data, mean_abs_shap, reshaped_data)
}
toc() # 35 s


biomarker_levels <- mean_abs_shap_met %>%
  filter(outcome == "cad") %>%
  arrange(desc(shap)) %>%
  pull(biomarker)

top_biomarkers <- mean_abs_shap_met %>%
  pivot_wider(names_from = outcome, values_from = shap) %>%
  pivot_longer(cols = -biomarker, names_to = "outcome", values_to = "shap") %>%
  group_by(outcome) %>%
  top_n(50, shap) %>%
  distinct(biomarker)
length(unique(top_biomarkers$biomarker)) # 62

dat <- mean_abs_shap_met %>%
  group_by(outcome) %>%
  mutate(shap = rescale(shap, c(0, 1))) %>% # Min-max normalization, equivalent to the mutate(shap = (shap - min(shap)) / (max(shap) - min(shap)))
  ungroup() %>%
  filter(biomarker %in% unique(top_biomarkers$biomarker)) %>%
  mutate(
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
    ),
    biomarker = factor(biomarker, levels = rev(biomarker_levels)),
    shap_truncated = ifelse(shap > 0.6, 0.6, shap)
  )
quantile(dat$shap, seq(0, 1, by = 0.1))
# quantile(dat %>% filter(outcome == "CAD") %>% pull(shap), seq(0, 1, by = 0.1))
# nrow(dat %>% filter(shap > 0.4))

#### Plotting ####
heatmap_met <- ggplot(dat, aes(x = outcome, y = biomarker, fill = shap_truncated)) +
  geom_tile(colour = "white") + 
  theme_void() +
  theme(
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_text(size = 12, colour = "black", hjust = 1, angle = 90),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 16, vjust = 1, angle = 90, face = "bold"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"),
    plot.title = element_text(size = 16, hjust = .5)
    ) +
  scale_fill_gradientn(
    colors = rev(met.brewer("Cassatt1")),
    # colours = c("#8282aa", "#9d9dc7", "#c9c9dd", "#edd7d9", "#e3aba7", "#d88782"),
    limit = c(0, 0.6),
    breaks = c(0, 0.3, 0.6),
    labels = c(0, 0.3, ">0.6"),
    name = "Normalized mean |SHAP|") +
  guides(
    fill = guide_colorbar(
      title.position = "top",
      title.hjust = 0.5
    )
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  )


```


#### Proteomics

```{r}

#### Data preparation ####
folder_path <- "your path/data/SHAP/Proteomics"

tic()
mean_abs_shap_pro <- tibble()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for (i in 1:9) {
  shap_data <- read_shap_data(folder_path, outcomes[i])
  
  # Compute the mean absolute SHAP value
  mean_abs_shap <- bind_rows(shap_data) %>%
    summarise(across(-c(fold, sample), ~ mean(abs(.))))
  
  # Convert the results into the required format
  reshaped_data <- tibble(
    biomarker = names(mean_abs_shap),
    outcome = outcomes[i],
    shap = as.numeric(mean_abs_shap[1, ])
  )

  # Merge into the final dataframe
  mean_abs_shap_pro <- bind_rows(mean_abs_shap_pro, reshaped_data)

  rm(shap_data, mean_abs_shap, reshaped_data)
}
toc() # 700 s


biomarker_levels <- mean_abs_shap_pro %>%
  filter(outcome == "cad") %>%
  arrange(desc(shap)) %>%
  pull(biomarker)

top_biomarkers <- mean_abs_shap_pro %>%
  pivot_wider(names_from = outcome, values_from = shap) %>%
  pivot_longer(cols = -biomarker, names_to = "outcome", values_to = "shap") %>%
  group_by(outcome) %>%
  top_n(50, shap) %>%
  distinct(biomarker)
length(unique(top_biomarkers$biomarker)) # 58

dat <- mean_abs_shap_pro %>%
  group_by(outcome) %>%
  mutate(shap = rescale(shap, c(0, 1))) %>% # Min-max normalization, equivalent to the mutate(shap = (shap - min(shap)) / (max(shap) - min(shap)))
  ungroup() %>%
  filter(biomarker %in% unique(top_biomarkers$biomarker)) %>%
  mutate(
    outcome = factor(
      outcome, 
      levels = c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death"),
      labels = c("CAD", "Stroke", "HF", "AF", "VA", "PAD", "AAA", "VTE", "CVDeath")
    ),
    biomarker = factor(biomarker, levels = rev(biomarker_levels)),
    # shap_truncated = ifelse(shap > 0.4, 0.4, shap)
    shap_truncated = ifelse(shap > 0.45, 0.45, shap)
  )
quantile(dat$shap, seq(0, 1, by = 0.1))
# nrow(dat %>% filter(shap > 0.4))

#### Plotting ####
heatmap_pro <- ggplot(dat, aes(x = outcome, y = biomarker, fill = shap_truncated)) +
  geom_tile(colour = "white") + 
  theme_void() +
  theme(
    axis.text.y = element_text(size = 12, colour = "black", hjust = 1, vjust = .5, angle = 0),
    axis.text.x = element_text(size = 12, colour = "black", hjust = 1, angle = 90),
    axis.title.x = element_text(size = 16, vjust = 0, angle = 00, face = "bold"),
    axis.title.y = element_text(size = 16, vjust = 1, angle = 90, face = "bold"),
    legend.title = element_text(size = 12, hjust = 0.1),
    legend.text = element_text(size = 12),
    legend.position = "bottom",
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm"),
    plot.title = element_text(size = 16, hjust = .5)
    ) +
  scale_fill_gradientn(
    colors = rev(met.brewer("Cassatt1")),
    # colours = c("#8282aa", "#9d9dc7", "#c9c9dd", "#edd7d9", "#e3aba7", "#d88782"),
    limit = c(0.15, 0.45),
    breaks = c(0.15, 0.3, 0.45),
    labels = c(0.15, "0.30", ">0.45"),
    name = "Normalized mean |SHAP|") +
  guides(
    fill = guide_colorbar(
      title.position = "top",
      title.hjust = 0.5
    )
  ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  )

```

#### Pool

```{r}

#### Pool ####
heatmap_met_wrap <- wrap_elements(
  heatmap_met +
  plot_annotation(
    title = "a",
    theme = theme(plot.title = element_text(size = 12, face = "bold"))
    )
  )

heatmap_pro_wrap <- wrap_elements(
  heatmap_pro +
  plot_annotation(
    title = "b",
    theme = theme(plot.title = element_text(size = 12, face = "bold"))
    )
  )

heatmap_pool <- wrap_elements(
  heatmap_met_wrap + heatmap_pro_wrap + 
    plot_layout(ncol = 2, widths = c(1, 0.95))
  )

```

## SHAP beeswarm plot

### Metabolomics

```{r}

#### Data preparation ####
folder_path <- "your path/data/SHAP/Metabolomics"

## Original da
# SHAP data
tic()
shap_met_list <- list()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for (i in 1:9) {
  shap_data <- read_shap_data(folder_path, outcomes[i])
  
  shap_met_list[[outcomes[i]]] <- bind_rows(shap_data)
  
  rm(shap_data)
}
toc() # 30 s

# X_test data
tic()
xtest_met <- read_xtest_data(folder_path) %>%
  bind_rows()
toc() # 45 s

# UK Biobank NMR information
load(str_c(getwd(), "/data/nmr_info.rda"))

# Data for plotting
subgroup_order <- c(
  "Amino acids",
  "Branched-chain amino acids",
  "Aromatic amino acids",
  "Fluid balance",
  "Inflammation",
  "Fatty acids",
  "Glycolysis related metabolites",
  "Ketone bodies",
  
  "Total lipids",
  "Cholesterol",
  "Free cholesterol",
  "Cholesteryl esters",
  "Phospholipids",
  "Triglycerides",
  "Other lipids",
 
  "Lipoprotein particle sizes",
  "Lipoprotein particle concentrations",
  "Chylomicrons and extremely large VLDL",
  "Very large VLDL",
  "Large VLDL",
  "Medium VLDL",
  "Small VLDL",
  "Very small VLDL",

  "Large LDL",
  "Medium LDL",
  "Small LDL",
  "IDL",
  "Very large HDL",
  "Large HDL",
  "Medium HDL",
  "Small HDL",
  "Apolipoproteins"
  )

## X_test
dat_xtest <- xtest_met %>%
  pivot_longer(cols = -c(fold, sample), names_to = "biomarker", values_to = "level")

## SHAP
# To reduce the dots, we divide the SHAP values into 100 quantiles for each biomarker
# Then, we calculate the mean SHAP value and level for each quantile
# The number of dots for each outcome is 168 * 100 = 16800
# Ref: 
# - https://www.nature.com/articles/s41591-022-01980-3/figures/5 (legend)
# - https://github.com/YanLuoCityU/MetabolomicsCommonDiseases/blob/main/2_analysis/Figure6_Attributions_3_Suns.ipynb
shap_met_data <- shap_met_list %>%
  imap_dfr(~ .x %>% mutate(outcome = .y))

dat_shap <- shap_met_data %>%
  pivot_longer(cols = -c(fold, sample, outcome), names_to = "biomarker", values_to = "shap") %>%
  as.data.table() %>%
  left_join(dat_xtest, by = c("fold", "sample", "biomarker")) %>%
  merge(., nmr_info %>% select(Biomarker, Group, Sub.Group), by.x = "biomarker", by.y = "Biomarker", all.x = TRUE) %>%
  mutate(
    Sub.Group = ifelse(Sub.Group == "", Group, Sub.Group),
    Sub.Group = sub("\\s*\\(.*\\)", "", Sub.Group),
    Sub.Group = factor(Sub.Group, levels = subgroup_order)
  ) %>%
  # group_by(fold, biomarker) %>%
  group_by(outcome, biomarker) %>%
  mutate(shap_quantile = ntile(shap, 100)) %>%
  ungroup()

# Text
dat_text <- dat_shap %>%
  group_by(biomarker) %>%
  slice(1) %>%
  ungroup() %>%
  select(biomarker, Sub.Group) %>%
  arrange(Sub.Group, biomarker) %>%
  group_by(Sub.Group) %>%
  do(bind_rows(., tibble(biomarker = "", Sub.Group = unique(.$Sub.Group)))) %>% # Use add_row() to insert an empty row after the last element of each group
  ungroup() %>%
  mutate(id = row_number() + 1) # Add an id column

dat_text_v1 <- dat_text %>%
  filter(biomarker != "")

dat_text_v2 <- dat_text %>%
  filter(biomarker != "") %>%
  group_by(Sub.Group) %>%
  summarise(id = mean(id)) %>%
  ungroup() %>%
  mutate(
    Sub.Group = case_when(
      Sub.Group == "Branched-chain amino acids" ~ "Branched-chain",
      Sub.Group == "Glycolysis related metabolites" ~ "Glycolysis related",
      Sub.Group == "Lipoprotein particle sizes" ~ "Lipoprotein particle ",
      Sub.Group == "Lipoprotein particle concentrations" ~ "Lipoprotein particle",
      Sub.Group == "Chylomicrons and extremely large VLDL" ~ "Chylomicrons and",
      TRUE ~ Sub.Group
    )
  ) %>%
  bind_rows(.,
    tibble(
      Sub.Group = c("amino acids", "metabolites", "sizes", "concentrations", "extremely large VLDL"),
      id = c(9.5, 32.5, 80, 84.2, 91)
      )
    )

dat_subgroup_lines <- dat_text_v1 %>% 
  filter(biomarker != "") %>%
  group_by(Sub.Group) %>%
  dplyr::summarize(start_id = min(id), end_id = max(id)) %>%
  mutate(
    start_id = ifelse(Sub.Group == "Inflammation", start_id - 0.25, start_id),
    end_id = ifelse(Sub.Group == "Inflammation", end_id + 0.25, end_id)
  )

# Final
dat <- dat_shap %>%
  group_by(outcome, biomarker, Sub.Group, shap_quantile) %>%
  summarise(
    mean_shap = mean(shap), 
    mean_level = mean(level)
    ) %>%
  ungroup() %>%
  left_join(dat_text %>% select(-Sub.Group), by = "biomarker") %>%
  mutate(
    mean_level_truncated = case_when(
      mean_level < -1.2 ~ -2,
      mean_level > 1.2 ~ 2,
      TRUE ~ mean_level
      )
    ) %>%
  group_by(outcome) %>%
  mutate(    
    jitter_factor = 1 - rescale(abs(mean_shap), to = c(0, 1)),
    id_jitter = id + runif(n = 168*100, min = -0.2 * jitter_factor, max = 0.2 * jitter_factor)
    ) %>%
  ungroup()

# color
paletteer_c("ggthemes::Red-Blue-White Diverging", 9)

#### Plotting ####
# CAD
beeswarm_met_cad <- ggplot(dat %>% filter(outcome == "cad")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.27), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.54), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.26, yend = 0.26), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.53, yend = 0.53), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.45,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-0.9, 0.9)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "CAD", x = -Inf, y = -0.9, size = 8, colour = "black") # ylim minimum


ggsave(str_c(getwd(), "/results/Figure/Beeswarm_Metabolomics_CAD.png"), plot = beeswarm_met_cad, width = 9, height = 9, unit = "in", dpi = 300, device = "png")


# Stroke
beeswarm_met_stroke <- ggplot(dat %>% filter(outcome == "stroke")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.27), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.54), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.26, yend = 0.26), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.53, yend = 0.53), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.45,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-0.9, 0.9)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "Stroke", x = -Inf, y = -0.8, size = 8, colour = "black")

# HF
beeswarm_met_hf <- ggplot(dat %>% filter(outcome == "hf")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.36), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.72), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.35, yend = 0.35), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.71, yend = 0.71), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.6,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-1.2, 1.2)) + # 27 points (SHAP <-0.8 or >0.8) have been removed for visualization
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "HF", x = -Inf, y = -1.2, size = 8, colour = "black")

# AF
beeswarm_met_af <- ggplot(dat %>% filter(outcome == "af")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.27), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.54), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.26, yend = 0.26), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.53, yend = 0.53), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.45,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-0.9, 0.9)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "AF", x = -Inf, y = -0.9, size = 8, colour = "black")

# VA
beeswarm_met_va <- ggplot(dat %>% filter(outcome == "va")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.42), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.84), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.41, yend = 0.41), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.83, yend = 0.83), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.7,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-1.4, 1.4)) + 
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "VA", x = -Inf, y = -1.4, size = 8, colour = "black")

# PAD
beeswarm_met_pad <- ggplot(dat %>% filter(outcome == "pad")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.45), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.9), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.44, yend = 0.44), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.89, yend = 0.89), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.75,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-1.5, 1.5)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "PAD", x = -Inf, y = -1.5, size = 8, colour = "black")

# AAA
beeswarm_met_aaa <- ggplot(dat %>% filter(outcome == "aaa")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.66), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 1.32), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.65, yend = 0.65), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 1.31, yend = 1.31), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -1.1,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-2.2, 2.2)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "AAA", x = -Inf, y = -2.2, size = 8, colour = "black")

# VTE
beeswarm_met_vte <- ggplot(dat %>% filter(outcome == "vt")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.27), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.54), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.26, yend = 0.26), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.53, yend = 0.53), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.45,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-0.9, 0.9)) + 
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "VTE", x = -Inf, y = -0.9, size = 8, colour = "black")

# CVDeath
beeswarm_met_cvdeath <- ggplot(dat %>% filter(outcome == "cvd_death")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1, 
    aes(x = id, label = biomarker, y = 0.42), # ylim maximum*0.3
    color = "black", size = 1.5, 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2, 
    aes(x = id, label = Sub.Group, y = 0.84), # ylim maximum*0.6
    color = "black", size = 2, fontface = "bold", 
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.41, yend = 0.41), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.83, yend = 0.83), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.7,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 200.5)) +
  scale_y_continuous(limits = c(-1.4, 1.4)) + # 27 points (SHAP <-0.8 or >0.8) have been removed for visualization
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-2, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 2))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "CVDeath", x = -Inf, y = -1.4, size = 8, colour = "black")

```

### Proteomics

```{r}

#### Data preparation ####
folder_path <- "your path/data/SHAP/Proteomics"

tic()
shap_pro_list <- list()
outcomes <- c("cad", "stroke", "hf", "af", "va", "pad", "aaa", "vt", "cvd_death")
for (i in 1:9) {
  shap_data <- read_shap_data(folder_path, outcomes[i])
  
  shap_pro_list[[outcomes[i]]] <- bind_rows(shap_data)
  
  rm(shap_data)
}
toc() # 500 s


# X_test data
tic()
xtest_pro <- read_xtest_data(folder_path) %>%
  bind_rows()
toc() # 45 s

# UK Biobank Olink information
olink_assay <- read_delim(str_c(getwd(), "/data/olink_assay.dat.txt"), show_col_types = F, quote = "\t") #%>%
  # mutate(Panel = str_replace(Panel, "II", "Ⅱ"))

# Data for plotting
subgroup_order = c(
  "Cardiometabolic", "Cardiometabolic II", "Inflammation", "Inflammation II",
  "Neurology", "Neurology II", "Oncology", "Oncology II"
  )

## X_test
dat_xtest <- xtest_pro %>%
  pivot_longer(cols = -c(fold, sample), names_to = "biomarker", values_to = "level")

## SHAP
### Top 150 biomarkers
top_biomarkers <- mean_abs_shap_pro %>%
  pivot_wider(names_from = outcome, values_from = shap) %>%
  pivot_longer(cols = -biomarker, names_to = "outcome", values_to = "shap") %>%
  group_by(outcome) %>%
  top_n(150, shap) %>%
  distinct(biomarker)
length(unique(top_biomarkers$biomarker)) # 175

shap_pro_data <- shap_pro_list %>%
  imap_dfr(~ .x %>% mutate(outcome = .y))

dat_shap <- shap_pro_data %>%
  select(c("fold", "sample", "outcome", unique(top_biomarkers$biomarker))) %>%
  pivot_longer(cols = -c(fold, sample, outcome), names_to = "biomarker", values_to = "shap") %>%
  as.data.table() %>%
  left_join(dat_xtest, by = c("fold", "sample", "biomarker")) %>%
  merge(., olink_assay %>% select(Assay, Panel), by.x = "biomarker", by.y = "Assay", all.x = TRUE) %>%
  mutate(Panel = factor(Panel, levels = subgroup_order)) %>%
  group_by(outcome, biomarker) %>%
  mutate(shap_quantile = ntile(shap, 100)) %>%
  ungroup()

# Text
dat_text <- dat_shap %>%
  group_by(biomarker) %>%
  slice(1) %>%
  ungroup() %>%
  select(biomarker, Panel) %>%
  arrange(Panel, biomarker) %>%
  group_by(Panel) %>%
  do(bind_rows(., tibble(biomarker = "", Panel = unique(.$Panel)))) %>% # Use add_row() to insert an empty row after the last element of each group
  ungroup() %>%
  mutate(id = row_number() + 1) # Add an id column


dat_text_v1 <- dat_text %>%
  filter(biomarker != "")

dat_text_v2 <- dat_text %>%
  filter(biomarker != "") %>%
  group_by(Panel) %>%
  summarise(id = mean(id)) %>%
  ungroup()

dat_subgroup_lines <- dat_text_v1 %>%
  filter(biomarker != "") %>%
  group_by(Panel) %>%
  dplyr::summarize(start_id = min(id), end_id = max(id))

# Final
dat <- dat_shap %>%
  group_by(outcome, biomarker, Panel, shap_quantile) %>%
  summarise(
    mean_shap = mean(shap), 
    mean_level = mean(level)
    ) %>%
  ungroup() %>%
  left_join(dat_text %>% select(-Panel), by = "biomarker") %>%
  mutate(
    mean_level_truncated = case_when(
      mean_level < -1.2 ~ -4,
      mean_level > 1.2 ~ 4,
      TRUE ~ mean_level
      )
    ) %>%
  group_by(outcome) %>%
  mutate(    
    jitter_factor = 1 - rescale(abs(mean_shap), to = c(0, 1)),
    id_jitter = id + runif(n = 175*100, min = -0.2 * jitter_factor, max = 0.2 * jitter_factor)
    ) %>%
  ungroup()

dat_legend <- dat

# color
# paletteer_c("ggthemes::Red-Blue-White Diverging", 9)

#### Plotting ####
# CAD
beeswarm_pro_cad <- ggplot(dat %>% filter(outcome == "cad")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "CAD", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# Stroke
beeswarm_pro_stroke <- ggplot(dat %>% filter(outcome == "stroke")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "Stroke", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# HF
beeswarm_pro_hf <- ggplot(dat %>% filter(outcome == "hf")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "HF", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# AF
beeswarm_pro_af <- ggplot(dat %>% filter(outcome == "af")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "AF", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# VA
beeswarm_pro_va <- ggplot(dat %>% filter(outcome == "va")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "VA", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# PAD
beeswarm_pro_pad <- ggplot(dat %>% filter(outcome == "pad")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "PAD", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# AAA
beeswarm_pro_aaa <- ggplot(dat %>% filter(outcome == "aaa")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "AAA", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# VTE
beeswarm_pro_vte <- ggplot(dat %>% filter(outcome == "vt")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "VTE", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

# CVDeath
beeswarm_pro_cvdeath <- ggplot(dat %>% filter(outcome == "cvd_death")) +
  geom_point(
    aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated),
    size = 1, alpha = 0.5
  ) +
  geom_textpath( # Biomarker labels
    data = dat_text_v1,
    aes(x = id, label = biomarker, y = 0.09), # ylim maximum*0.3
    color = "black", size = 1.5,
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_textpath( # Subgroup labels
    data = dat_text_v2,
    aes(x = id, label = Panel, y = 0.18), # ylim maximum*0.6 
    color = "black", size = 2, fontface = "bold",
    angle = 90, hjust = 0, vjust = 0.5
    ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.08, yend = 0.08), # ylim maximum*0.3-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_segment(
    data = dat_subgroup_lines,
    aes(x = start_id, xend = end_id, y = 0.17, yend = 0.17), # ylim maximum*0.6-0.1
    color = "grey50", linewidth = 0.25
  ) +
  geom_hline(
    yintercept = 0,
    color = "grey50", linetype = "solid", linewidth = 0.5
    ) +
  geom_hline(
    yintercept = -0.15,
    color = "grey50", linetype = "dashed", linewidth = 0.5 # ylim minimum*0.5
    ) +
  coord_polar(clip = "off") +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "none",
    # panel.background = element_blank(),
    panel.grid  = element_blank(),
    plot.margin = unit(c(t = -3, r = -3, b = -3, l = -3), "cm")
  ) +
  scale_x_continuous(limits = c(0.5, 187.5)) +
  scale_y_continuous(limits = c(-0.3, 0.3)) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4))
    ) +
  labs(
    x = NULL,
    y = NULL,
    title = NULL
  ) +
  annotate("text", label = "CVDeath", x = -Inf, y = -0.3, size = 8, colour = "black") # ylim minimum

```


### Pool

```{r}

#### Legend ####
legend_1 <- ggplot(dat_legend %>% filter(outcome == "cad")) +
  geom_point(aes(x = id_jitter, y = mean_shap, colour = mean_level_truncated)) +
  theme_classic() +
  theme(
    axis.text.x = element_blank(),
    axis.text.y = element_blank(),
    axis.ticks = element_blank(),
    axis.line = element_blank(),
    legend.position = "bottom",
    panel.background = element_blank(),
    plot.margin = unit(c(t = 0, r = 0, b = 0, l = 0), "cm")
  ) +
  scale_colour_gradientn( # Biomarker levels (truncated)
    colors = rev(c("#A90C38FF", "#DB4446FF", "#F78171FF", "#FFBFB2FF", "#FFFFFFFF", "#B4D2E9FF", "#7BA8CCFF", "#5180ACFF", "#2E5A87FF")),
    values = rescale(c(-4, -1, -0.75, -0.5, -0.25, 0, 0.25, 0.5, 0.75, 1, 4)),
    breaks = c(-4, 4),
    labels = c("Low", "High"),
    name = "Biomarker value"
    ) +
  guides(
    colour = guide_colorbar(
      title.position = "top",
      title.hjust = 0.5
    )
  )


#### Pool ####
# CAD
beeswarm_met_wrap <- wrap_elements(
  beeswarm_met_cad +
  plot_annotation(
    title = "c",
    theme = theme(plot.title = element_text(size = 12, face = "bold"))
    )
  )

beeswarm_pro_wrap <- wrap_elements(
  beeswarm_pro_cad +
  plot_annotation(
    title = "d",
    theme = theme(plot.title = element_text(size = 12, face = "bold"))
    )
  )

beeswarm_pool <- wrap_elements(
  beeswarm_met_wrap + beeswarm_pro_wrap + 
    wrap_elements(ggpubr::get_legend(legend_1)) +
    plot_layout(nrow = 3, heights = c(1, 1, 0.1))
  )


dat <- data.frame(
  x = c(0.5, 0.75),
  y = c(1, 0.75),
  xend = c(1, 0.95),
  yend = c(0.6, 0.95)
)

legend_2 <- ggplot() +
  geom_curve(
    data = dat[1,],
    aes(x = x, y = y, xend = xend, yend = yend),
    curvature = -0.4, color = "black"
  ) +
  geom_segment(
    data = dat[2,],
    aes(x = x, y = y, xend = xend, yend = yend),
    arrow = arrow(length = unit(0.2, "cm")),
    color = "black"
  ) +
  geom_segment(
    data = dat[2,],
    aes(x = xend, y = yend, xend = x, yend = y),
    arrow = arrow(length = unit(0.2, "cm")),
    color = "black"
  ) +
  annotate("text", x = 0.2, y = 1.1, label = "Positive contribution", size = 3, hjust = 0) +
  annotate("text", x = 0.15, y = 0.6, label = "Negative contribution", size = 3, hjust = 0) +
  theme_void()

beeswarm_pool <- beeswarm_pool +
  inset_element(
    legend_2,
    left = 0.75,
    right = 0.9,
    top = 0.53,
    bottom = 0.48
  )

# Metabolomics for other diseases
dat <- data.frame(
  x = c(0.5, 0.7),
  y = c(1, 0.6),
  xend = c(1, 0.9),
  yend = c(0.5, 1.2)
)

legend_2 <- ggplot() +
  geom_curve(
    data = dat[1,],
    aes(x = x, y = y, xend = xend, yend = yend),
    curvature = -0.4, color = "black"
  ) +
  geom_segment(
    data = dat[2,],
    aes(x = x, y = y, xend = xend, yend = yend),
    arrow = arrow(length = unit(0.2, "cm")),
    color = "black"
  ) +
  geom_segment(
    data = dat[2,],
    aes(x = xend, y = yend, xend = x, yend = y),
    arrow = arrow(length = unit(0.2, "cm")),
    color = "black"
  ) +
  scale_x_continuous(limits = c(0.5, 1)) +
  scale_y_continuous(limits = c(0.5, 1.2)) +
  annotate("text", x = 0.7, y = 1.2, label = "Positive contribution", size = 3, hjust = 0) +
  annotate("text", x = 0.53, y = 0.6, label = "Negative contribution", size = 3, hjust = 0) +
  theme_void()

beeswarm_met_all <- wrap_elements(
  beeswarm_met_stroke + beeswarm_met_hf + beeswarm_met_af + 
    beeswarm_met_va + beeswarm_met_pad + beeswarm_met_aaa + 
    beeswarm_met_vte + beeswarm_met_cvdeath +
  plot_layout(nrow = 3) +
  plot_annotation(
    tag_levels = "a",
    theme = theme(plot.tag = element_text(face = "bold"))
    )
  )

beeswarm_met_all <- beeswarm_met_all +
  inset_element(
    legend_2,
    left = 0.7,
    right = 0.85,
    top = 0.2,
    bottom = 0.1
  )

beeswarm_met_pool <- wrap_elements(
  beeswarm_met_all + wrap_elements(ggpubr::get_legend(legend_1)) +
    plot_layout(nrow = 2, heights = c(1, 0.05))
  )

# Proteomics for other diseases
beeswarm_pro_all <- wrap_elements(
  beeswarm_pro_stroke + beeswarm_pro_hf + beeswarm_pro_af + 
    beeswarm_pro_va + beeswarm_pro_pad + beeswarm_pro_aaa + 
    beeswarm_pro_vte + beeswarm_pro_cvdeath +
  plot_layout(nrow = 3) +
  plot_annotation(
    tag_levels = "a",
    theme = theme(plot.tag = element_text(face = "bold"))
    )
  )

beeswarm_pro_all <- beeswarm_pro_all +
  inset_element(
    legend_2,
    left = 0.7,
    right = 0.85,
    top = 0.2,
    bottom = 0.1
  )

beeswarm_pro_pool <- wrap_elements(
  beeswarm_pro_all + wrap_elements(ggpubr::get_legend(legend_1)) +
    plot_layout(nrow = 2, heights = c(1, 0.05))
  )

ggsave("your path/results/Figure/Beeswarm_pro_pool.pdf", plot = beeswarm_pro_pool, width = 27, height = 24, unit = "in", dpi = 300, device = "pdf")

```

## Pool

```{r}

design <- "
AABB
"

shap_pool <- wrap_plots(
  heatmap_pool, beeswarm_pool,
  design = design
  )

ggsave(str_c(getwd(), "/results/Figure/Model_interpretation.pdf"), plot = shap_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "pdf")

ggsave(str_c(getwd(), "/results/Figure/Model_interpretation.png"), plot = shap_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "png")

ggsave(str_c(getwd(), "/results/Figure/Model_interpretation.tiff"), plot = shap_pool, width = 16, height = 16, unit = "in", dpi = 300, device = "tiff")

```
